## Discussion

<!-- Discuss why vg is doing better -->
Overall, vg was the most accurate SV genotyper in our benchmarks.
These results show that variant calling benefits from variant-aware read mapping and graph based genotyping, a finding consistent with previous studies[@tag:vgnbt; @tag:bakeoff; @doi:10.1038/s41588-018-0316-4; @tag:graphtyper; @tag:bayestyper].
We took advantage of newly released datasets for our evaluation, which feature up to 3.7 times more variants than the more widely-used GIAB benchmark.
More and more large-scale projects are using low cost short-read technologies to sequence the genomes of thousands to hundreds of thousands of individuals (e.g. the Pancancer Analysis of Whole Genomes[@url:https://dcc.icgc.org/pcawg], the Genomics England initiative[@url:https://www.genomicsengland.co.uk], and the TOPMed consortium[@url:https://www.nhlbiwgs.org/]).
We believe pangenome graph-based approaches will improve both how efficiently SVs can be represented, and how accurately they can be genotyped with this type of data.

<!-- Input data quality: "sequence-resolved", break-point fine-tuning. -->
A particular advantage of our method is that it does not require exact breakpoint resolution in the variant library.
Our simulations showed that vg's SV genotyping algorithm is robust to errors of as much as 10 bp in breakpoint location.
However, there is an upper limit to this flexibility, and we find that vg cannot accurately genotype variants with much higher uncertainty in the breakpoint location (like those discovered through read coverage analysis).
vg is also capable of fine-tuning SV breakpoints by augmenting the graph with differences observed in read alignments.
Simulations showed that this approach can usually correct small errors in SV breakpoints (Figure {@fig:simerror-bkpt} and Table {@tbl:simerror-bkpt}).

<!-- Already superior but will only get better with new vg dev -->
vg uses a unified framework to call and score different variant types simultaneously.
In this work, we only considered graphs containing certain types of SVs, but the same methods can be extended to a broader range of graphs.
For example, we are interested in evaluating how genotyping SVs together with SNPs and small indels using a combined graph effects the accuracy of studying either alone.
The same methods used for genotyping known variants in this work can also be extended to call novel variants by first augmenting the graph with edits from the mapped reads.
This approach, which was used only in the breakpoint fine-tuning portion of this work, could be further used to study small variants around and nested within SVs.
Novel SVs could be called by augmenting the graph with long-read mappings.
vg is entirely open source, and its ongoing development is supported by a growing community of researchers and users with common interest in scalable, unbiased pangenomic analyses and representation.
We expect this collaboration to continue to foster increases in the speed, accuracy and applicability of methods based on pangenome graphs in the years ahead.

<!-- Benefits of *de novo* assemblies -->
Our results suggest that constructing a graph from *de novo* assembly alignment instead of a VCF leads to better SV genotyping.
High quality *de novo* assemblies for human are becoming more and more common due to improvements in technologies like optimized mate-pair libraries[@tag:denmark] and long-read sequencing[@doi:10.1038/nbt.4060].
We expect future graphs to be built from the alignment of numerous *de novo* assemblies, and we are presently working on scaling our assembly-based pipeline to human-sized genome assemblies.
Another challenge is creating genome graphs that integrate assemblies with variant-based data resources.
One possible approach is to progressively align assembled contigs into variation graphs constructed from variant libraries, but methods for doing so are still experimental.

## Conclusion

In this study, the vg toolkit was compared to existing SV genotypers across several high-quality SV catalogs.
We showed that its method of mapping reads to a variation graph leads to better SV genotyping compared to other state of the art methods.
This work introduces a flexible strategy to integrate the growing number of SVs being discovered with higher resolution technologies into a unified framework for genome inference.
Our work on whole genome alignment graphs shows the benefit of directly utilizing *de novo* assemblies rather than variant catalogs to integrate SVs in genome graphs.
We expect this latter approach to increase in significance as the reduction in long read sequencing costs drives the creation of numerous new *de novo* assemblies.
We envision a future in which the lines between variant calling, alignment, and assembly are blurred by rapid changes in sequencing technology.
Fully graph based approaches, like the one we present here, will be of great utility in this new phase of genome inference.
