## Declarations

### Availability of data and material

The commands used to run the analyses presented in this study are available at 
[github.com/vgteam/sv-genotyping-paper](https://github.com/vgteam/sv-genotyping-paper) under a MIT license. 
The version of the code used in this manuscript has been deposited in the following Zenodo repository: 
https://doi.org/10.5281/ZENODO.3517248[@raw:repo].
The scripts to generate the manuscript, including figures and tables, are available at [github.com/jmonlong/manu-vgsv](https://github.com/jmonlong/manu-vgsv).

The simulated reference sequence and reads were deposited at 
[https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=vgsv2019/simulation](https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=vgsv2019/simulation).

The sequencing data for HG00514, HG00733, and NA19240 was downloaded from SRA: 
[ERR903030](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=ERR903030), 
[ERR894724](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=ERR894724), 
[ERR894724](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=ERR894724).

The haplotype-resolved SVs from the Human Genome Structural Variation Consortium[@tag:hgsvc] were deposited on the GitHub repository[@raw:repo]: 
[https://github.com/vgteam/sv-genotyping-paper/tree/master/human/hgsvc](https://github.com/vgteam/sv-genotyping-paper/tree/master/human/hgsvc).
The simulated reads for HG00514 using the HGSVC genotypes were deposited at 
[https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=vgsv2019/sim-HG00514-30x.fq.gz](https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=vgsv2019/sim-HG00514-30x.fq.gz).

The Genome in a bottle catalog[@doi:10.1101/664623] was downloaded from 
[ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_UnionSVs_12122017/svanalyzer_union_171212_v0.5.0_annotated.vcf.gz](ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/analysis/NIST_UnionSVs_12122017/svanalyzer_union_171212_v0.5.0_annotated.vcf.gz).
Sequencing data for HG002 came from 
[ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/NIST_HiSeq_HG002_Homogeneity-10953946/NHGRI_Illumina300X_AJtrio_novoalign_bams/HG002.hs37d5.300x.bam](ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/AshkenazimTrio/HG002_NA24385_son/NIST_HiSeq_HG002_Homogeneity-10953946/NHGRI_Illumina300X_AJtrio_novoalign_bams/HG002.hs37d5.300x.bam).

The SV catalog from Audano et al.[@tag:audano2019] was prepared from 
[http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/hgsv_sv_discovery/working/20181025_EEE_SV-Pop_1/VariantCalls_EEE_SV-Pop_1/EEE_SV-Pop_1.ALL.sites.20181204.vcf.gz](http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/hgsv_sv_discovery/working/20181025_EEE_SV-Pop_1/VariantCalls_EEE_SV-Pop_1/EEE_SV-Pop_1.ALL.sites.20181204.vcf.gz).

The genotype predictions for each method benchmarked are available at 
[https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=vgsv2019/vcfs/](https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=vgsv2019/vcfs/)

The yeast assemblies were downloaded from 
[https://yjx1217.github.io/Yeast_PacBio_2016/data/](https://yjx1217.github.io/Yeast_PacBio_2016/data/).
The sequencing reads were downloaded from SRA: 
[SRR4074413](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074413),
[SRR4074412](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074412),
[SRR4074411](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074411),
[SRR4074394](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074394),
[SRR4074385](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074385),
[SRR4074384](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074384),
[SRR4074383](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074383),
[SRR4074358](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074358),
[SRR4074258](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074258),
[SRR4074257](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074257),
[SRR4074256](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074256),
[SRR4074255](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR4074255).

### Competing interests

The authors declare that they have no competing interests.

### Ethics approval and consent to participate

Not applicable.

### Funding

Research reported in this publication was supported by the National Human Genome Research Institute of the National Institutes of Health under Award Number U54HG007990 and U01HL137183. 
This publication was supported by a Subagreement from European Molecular Biology Laboratory with funds provided by Agreement No. 2U41HG007234 from National Institute of Health, NHGRI. 
Its contents are solely the responsibility of the authors and do not necessarily represent the official views of National Institute of Health, NHGRI or European Molecular Biology Laboratory.
The research was made possible by the generous financial support of the W.M. Keck Foundation (DT06172015).

JAS was further supported by the Carlsberg Foundation. 
DH was supported by the International Max Planck Research School for Computational Biology and Scientific Computing doctoral program. 
JE was supported by the Jack Baskin and Peggy Downes-Baskin Fellowship. 
AMN was supported by the National Institutes of Health (5U41HG007234), the W.M. Keck Foundation (DT06172015) and the Simons Foundation (SFLIFE# 35190). 


### Authors' contributions

EG, AN, GH, JS, JE and ED implemented the read mapping and variant calling in the vg toolkit.
GH, DH, JM, JAS and EG performed analysis on the different datasets.
GH, DH, JM and BP designed the study.
GH, DH and JM drafted the manuscript.
All authors read, reviewed, and approved the final manuscript.


### Acknowledgements

We thank Peter Audano for sharing the CHMPD dataset and for his assistance with SMRT-SV v2.

### Authors' information

These authors contributed equally: Glenn Hickey, David Heller, Jean Monlong.

<!-- ## Supplementary information -->

<!-- Additional file 1: Table S1-S6, Figure S1-S18, Supplementary information. -->
