<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Glenn Hickey" />
  <meta name="author" content="David Heller" />
  <meta name="author" content="Jean Monlong" />
  <meta name="author" content="Jonas Andreas Sibbesen" />
  <meta name="author" content="Jouni Siren" />
  <meta name="author" content="Jordan Eizenga" />
  <meta name="author" content="Eric T. Dawson" />
  <meta name="author" content="Erik Garrison" />
  <meta name="author" content="Adam Novak" />
  <meta name="author" content="Benedict Paten" />
  <meta name="dcterms.date" content="2019-10-16" />
  <meta name="keywords" content="structural variation, pangenome, variant genotyping" />
  <title>Genotyping structural variants in pangenome graphs using the vg toolkit</title>
  <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genotyping structural variants in pangenome graphs using the vg toolkit</h1>
</header>
<p><small><em> This manuscript (<a href="https://jmonlong.github.io/manu-vgsv/v/c42f04a9baef74bad29efeedf1e359970e3d286a/">permalink</a>) was automatically generated from <a href="https://github.com/jmonlong/manu-vgsv/tree/c42f04a9baef74bad29efeedf1e359970e3d286a">jmonlong/manu-vgsv@c42f04a</a> on October 16, 2019. </em></small></p>
<h2 id="authors">Authors</h2>
<p>Glenn Hickey<sup>1,☯</sup>, <a href="https://orcid.org/0000-0001-8346-9565"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> David Heller<sup>1,2,☯</sup>, <a href="https://orcid.org/0000-0002-9737-5516"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Jean Monlong<sup>1,☯</sup>, <a href="https://orcid.org/0000-0002-5528-0236"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Jonas Andreas Sibbesen<sup>1</sup>, <a href="https://orcid.org/0000-0001-5828-4139"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Jouni Siren<sup>1</sup>, <a href="https://orcid.org/0000-0001-8345-8356"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Jordan Eizenga<sup>1</sup>, <a href="https://orcid.org/0000-0001-5448-1653"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Eric T. Dawson<sup>3,4</sup>, <a href="https://orcid.org/0000-0003-3821-631X"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Erik Garrison<sup>1</sup>, <a href="https://orcid.org/0000-0001-5828-047X"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Adam Novak<sup>1</sup>, <a href="https://orcid.org/0000-0001-8863-3539"><img src="images/orcid.svg" alt="ORCID icon" class="inline_icon" /></a> Benedict Paten<sup>1,†</sup></p>
<p><sup>☯</sup> — These authors contributed equally to this work</p>
<p><sup>†</sup> — To whom correspondence should be addressed: bpaten@ucsc.edu <small></p>
<ol type="1">
<li>UC Santa Cruz Genomics Institute, University of California, Santa Cruz, California, USA</li>
<li>Max Planck Institute for Molecular Genetics, Berlin, Germany</li>
<li>Department of Genetics, University of Cambridge, Cambridge, UK</li>
<li>Division of Cancer Epidemiology and Genetics, National Cancer Institute, Rockville, Maryland, USA</li>
</ol>
<p></small></p>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p>Structural variants (SVs) are significant components of genetic diversity and have been associated with diseases, but the technological challenges surrounding their representation and identification make them difficult to study relative to point mutations. Still, thousands of SVs have been characterized, and catalogs continue to improve with new technologies. In parallel, variation graphs have been proposed to represent human pangenomes, offering reduced reference bias and better mapping accuracy than linear reference genomes. We contend that variation graphs provide an effective means for leveraging SV catalogs for short-read SV genotyping experiments. In this work, we extend vg (a software toolkit for working with variation graphs) to support SV genotyping. We show that it is capable of genotyping insertions, deletions and inversions, even in the presence of small errors in the location of the SVs breakpoints. We then benchmark vg against state-of-the-art SV genotypers using three high-quality sequence-resolved SV catalogs generated by recent studies ranging up to 97,368 variants in size. We find that vg systematically produces the best genotype predictions in all datasets. In addition, we use assemblies from 12 yeast strains to show that graphs constructed directly from aligned <em>de novo</em> assemblies can improve genotyping compared to graphs built from intermediate SV catalogs in the VCF format. Our results demonstrate the power of variation graphs for SV genotyping. Beyond single nucleotide variants and short insertions/deletions, the vg toolkit now incorporates SVs in its unified variant calling and genotyping framework and provides a natural solution to integrate high-quality SV catalogs and assemblies.</p>
<h2 id="introduction" class="page_break_before">Introduction</h2>
<p>A structural variant (SV) is a genomic mutation involving 50 or more base pairs. SVs can take several forms such as deletions, insertions, inversions, translocations or other complex events.<br />
Due to their greater size, SVs often have a larger impact on phenotype than smaller events such as single nucleotide variants (SNVs) and small insertions and deletions (indels)<span class="citation" data-cites="2gpKwL67">[<a href="#ref-2gpKwL67">1</a>]</span>. Indeed, SVs have long been associated with developmental disorders, cancer and other complex diseases and phenotypes<span class="citation" data-cites="ebc66eBr">[<a href="#ref-ebc66eBr">2</a>]</span>.</p>
<p>Despite their importance, SVs remain much more poorly studied than their smaller mutational counterparts. This discrepancy stems from technological limitations. Short read sequencing has provided the basis of most modern genome sequencing studies due to its high base-level accuracy and relatively low cost, however, it is poorly suited for discovering SVs. The central obstacle is in mapping short reads to the human reference genome. It is generally difficult or impossible to unambiguously map a short read if the sample whose genome is being analyzed differs substantially from the reference at the read’s location. The large size of SVs virtually guarantees that short reads derived from them will not map to the linear reference genome. For example, if a read corresponds to sequence in the middle of a large reference-relative insertion, then there is no location in the reference that corresponds to a correct mapping. The best result a read mapper could hope to produce would be to leave it unmapped. Moreover, SVs often lie in repeat-rich regions, which further frustrate read mapping algorithms.</p>
<p>Short reads can be more effectively used to genotype known SVs. This is important, as even though efforts to catalog SVs with other technologies have been highly successful, their cost currently prohibits their use in large-scale studies that require hundreds or thousands of samples such as disease association studies. Traditional SV genotypers start from reads that were mapped to a reference genome, extracting aberrant mapping that might support the presence of the SV of interest. Current methods such as SVTyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">3</a>]</span> and Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">4</a>]</span> typically focus on split reads and paired reads mapped too close or too far from each other. These discordant reads are tallied and remapped to the reference sequence modified with the SV of interest in order to genotype deletions, insertions, duplications, inversions and translocations. SMRT-SV v2 Genotyper (henceforth referred to as SMRT-SV v2) uses a different approach: the reference genome is augmented with SV-containing sequences as alternate contigs and the resulting mappings are evaluated with a machine learning model trained for this purpose<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>.</p>
<p>The catalog of known SVs in human is quickly expanding. Several large-scale projects have used short-read sequencing and extensive discovery pipelines on large cohorts, compiling catalogs with tens of thousands of SVs in humans<span class="citation" data-cites="qA6dWFP py6BC5kj">[<a href="#ref-qA6dWFP">6</a>,<a href="#ref-py6BC5kj">7</a>]</span>, using split read and discordant pair based methods like SVTyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">3</a>]</span> and Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">4</a>]</span> to find SVs using short read sequencing. More recent studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which was novel and sequence-resolved<span class="citation" data-cites="z91V6jjU rs7e40wC PRx3qEIm 121OWcTA4 3NNFS6U2">[<a href="#ref-PRx3qEIm">10</a>,<a href="#ref-121OWcTA4">11</a>,<a href="#ref-3NNFS6U2">5</a>,<a href="#ref-z91V6jjU">8</a>,<a href="#ref-rs7e40wC">9</a>]</span>. These technologies are also enabling the production of high-quality <em>de novo</em> genome assemblies<span class="citation" data-cites="z91V6jjU 6KbgcueR">[<a href="#ref-6KbgcueR">12</a>,<a href="#ref-z91V6jjU">8</a>]</span>, and large blocks of haplotype-resolved sequences<span class="citation" data-cites="Pu6SY37C">[<a href="#ref-Pu6SY37C">13</a>]</span>. Such technical advances promise to expand the amount of known genomic variation in humans in the near future, and further power SV genotyping studies. Representing known structural variation in the wake of increasingly larger datasets poses a considerable challenge, however. VCF, the standard format for representing small variants, is unwieldy when used for SVs due its unsuitability for expressing nested or complex variants. Another strategy consists in incorporating SVs into a linear pangenome reference via alt contigs, but it also has serious drawbacks. Alt contigs tend to increase mapping ambiguity. In addition, it is unclear how to scale this approach as SV catalogs grow.</p>
<p>Pangenomic graph reference representations offer an attractive approach for storing genetic variation of all types<span class="citation" data-cites="Qa8mx6Ll">[<a href="#ref-Qa8mx6Ll">14</a>]</span>. These graphical data structures can seamlessly represent both SVs and point mutations using the same semantics. Moreover, including known variants in the reference makes read mapping, variant calling and genotyping variant-aware. This leads to benefits in terms of accuracy and sensitivity<span class="citation" data-cites="10jxt15v0 DuODeStx 11Jy8B61m">[<a href="#ref-10jxt15v0">15</a>,<a href="#ref-DuODeStx">16</a>,<a href="#ref-11Jy8B61m">17</a>]</span>. The coherency of this model allows different variant types to be called and scored simultaneously in a unified framework.</p>
<p>vg is the first openly available variation graph tool to scale to multi-gigabase genomes. It provides read mapping, variant calling and visualization tools<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">15</a>]</span>. In addition, vg can build graphs both from variant catalogs in the VCF format and from assembly alignments.</p>
<p>Other tools have used genome graphs or pangenomes to genotype variants. GraphTyper realigns mapped reads to a graph built from known SNVs and short indels using a sliding-window approach<span class="citation" data-cites="ohTIiqfV">[<a href="#ref-ohTIiqfV">18</a>]</span>. BayesTyper first builds a set of graphs from known variants including SVs, then genotypes variants by comparing the distribution of k-mers in the sequencing reads with the k-mers of haplotype candidate paths in the graph<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">19</a>]</span>. Paragraph builds a graph for each breakpoint of known variants <span class="citation" data-cites="zjxFCnET">[<a href="#ref-zjxFCnET">20</a>]</span>, then for each breakpoint, it pulls out all nearby reads from the linear alignment and re-aligns them to the graph. Genotypes are computed using the read coverage from the pair of breakpoint graphs corresponding to each SV. SMRT-SV v2 uses a different approach: the reference genome is augmented with SV-containing sequences as alternate contigs and the resulting mappings are evaluated with a machine learning model trained for this purpose<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. These graph-based approaches showed clear advantages over standard methods that use only the linear reference.</p>
<p>In this work, we present a SV genotyping framework based on the variation graph model and implemented in the vg toolkit. We show that this method is capable of genotyping known deletions, insertions and inversions, and that its performance is not inhibited by small errors in the specification of SV allele breakpoints. We evaluated the genotyping accuracy of our approach using simulated and real Illumina reads and a pangenome built from SVs discovered in recent long-read sequencing studies<span class="citation" data-cites="3NNFS6U2 vQTymKCj RCzDkVqI 16GvGhO20">[<a href="#ref-vQTymKCj">21</a>,<a href="#ref-RCzDkVqI">22</a>,<a href="#ref-16GvGhO20">23</a>,<a href="#ref-3NNFS6U2">5</a>]</span>, We also compared vg’s performance with state-of-the-art SV genotypers: SVTyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">3</a>]</span>, Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">4</a>]</span>, BayesTyper<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">19</a>]</span>, Paragraph<span class="citation" data-cites="zjxFCnET">[<a href="#ref-zjxFCnET">20</a>]</span> and SMRT-SV v2<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. Across the datasets we tested, which range in size from 26k to 97k SVs, vg is the best performing SV genotyper on real short-read data for all SV types in the majority of cases. Finally, we demonstrate that a pangenome graph built from the alignment of <em>de novo</em> assemblies of diverse <em>Saccharomyces cerevisiae</em> strains improves SV genotyping performance.</p>
<h2 id="results">Results</h2>
<h3 id="structural-variation-in-vg">Structural variation in vg</h3>
<p>We used vg to implement a straightforward SV genotyping pipeline. Reads are mapped to the graph and used to compute the read support for each node and edge (see <a href="#supplementary-information">Supplementary Information</a> for a description of the graph formalism). Sites of variation within the graph are then identified using the snarl decomposition as described in <span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">24</a>]</span>. These sites correspond to intervals along the reference paths (ex. contigs or chromosomes) which are embedded in the graph. They also contain nodes and edges deviating from the reference path, which represent variation at the site. For each site, the two most supported paths between its interval (haplotypes) are determined, and their relative supports used to produce a genotype at that site (Figure <a href="#fig:1">1</a>a). The pipeline is described in detail in <a href="#simulation-experiment">Methods</a>. We rigorously evaluated the accuracy of our method on a variety of datasets, and present these results in the remainder of this section.</p>
<figure>
<img src="images/panel1.png" alt="Figure 1: Structural variation in vg. a) vg uses the read coverage over possible paths to genotype variants in a bubble or more complex snarl. The cartoon depicts the case of an heterozygous insertion and an homozygous deletion. The algorithm is described in more details in Methods. b) Simulation experiment. Each subplot shows a comparison of genotyping accuracy for four SV calling methods. Results are separated between types of variation (insertions, deletions, and inversions). The experiments were also repeated with small random errors introduced to the VCF to simulate breakpoint uncertainty. For each experiment, the x-axis is the simulated read depth and the y-axis shows the maximum F1 across different minimum quality thresholds. SVTyper cannot genotype insertions, hence the missing line in the top panels." id="fig:1" /><figcaption><span>Figure 1:</span> <strong>Structural variation in vg.</strong> a) vg uses the read coverage over possible paths to genotype variants in a bubble or more complex snarl. The cartoon depicts the case of an heterozygous insertion and an homozygous deletion. The algorithm is described in more details in <a href="#toil-vg-call">Methods</a>. b) Simulation experiment. Each subplot shows a comparison of genotyping accuracy for four SV calling methods. Results are separated between types of variation (insertions, deletions, and inversions). The experiments were also repeated with small random errors introduced to the VCF to simulate breakpoint uncertainty. For each experiment, the x-axis is the simulated read depth and the y-axis shows the maximum F1 across different minimum quality thresholds. SVTyper cannot genotype insertions, hence the missing line in the top panels.</figcaption>
</figure>
<h3 id="simulated-dataset">Simulated dataset</h3>
<p>As a proof of concept, we simulated genomes and different types of SVs with a size distribution matching real SVs<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">21</a>]</span>. We compared vg against Paragraph, SVTyper, Delly, and BayesTyper across different levels of sequencing depth. We also added some errors (1-10bp) to the location of the breakpoints to investigate their effect on genotyping accuracy (see <a href="#simulation-experiment">Methods</a>). The results are shown in Figure <a href="#fig:1">1</a>b.</p>
<p>When using the correct breakpoints, most methods performed similarly, with differences only becoming visible at very low sequencing depths. Only vg and Paragraph maintained their performance in the presence of 1-10 bp errors in the breakpoint locations. The dramatic drop for BayesTyper can be explained by its k-mer-based approach that requires precise breakpoints. Overall, these results show that vg is capable of genotyping SVs and is robust to breakpoint inaccuracies in the input VCF.</p>
<h3 id="hgsvc-dataset">HGSVC dataset</h3>
<p>72,485 structural variants from The Human Genome Structural Variation Consortium (HGSVC) were used to benchmark the genotyping performance of vg against the four other SV genotyping methods. This high-quality SV catalog was generated from three samples using a consensus from different sequencing, phasing, and variant calling technologies<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">21</a>]</span>. The three individual samples represent different human populations: Han Chinese (HG00514), Puerto-Rican (HG00733), and Yoruban Nigerian (NA19240). We used these SVs to construct a graph with vg and as input for the other genotypers. Using short sequencing reads, the SVs were genotyped and compared with the genotypes in the original catalog (see <a href="#hgsvc-analysis">Methods</a>).</p>
<p>First we compared the methods using simulated reads for HG00514. This represents the ideal situation where the SV catalog exactly matches the SVs supported by the reads. BayesTyper and vg showed the best F1 score and precision-recall trade-offs (Figures <a href="#fig:2">2</a>a and <a href="#fig:hgsvc-sim-geno">S1</a>, Table <a href="#tbl:hgsvc">S1</a>), outperforming the other methods by a clear margin. When restricting the comparisons to regions not identified as tandem repeats or segmental duplications, the genotyping predictions were significantly better for all methods. We observed similar results when evaluating the presence of an SV call instead of the exact genotype (Figures <a href="#fig:2">2</a>a and <a href="#fig:hgsvc-sim">S2</a>).</p>
<figure>
<img src="images/panel2.png" alt="Figure 2: Structural variants from the HGSVC and Genome in a Bottle datasets. HGSVC: Simulated and real reads were used to genotype SVs and compared with the high-quality calls from Chaisson et al.[21]. Reads were simulated from the HG00514 individual. Using real reads, the three HG00514, HG00733, and NA19240 individuals were tested. GIAB: Real reads from the HG002 individual were used to genotype SVs and compared with the high-quality calls from the Genome in a Bottle consortium[22,23,25]. a) Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Results are separated across panels by variant type: insertions and deletions. SVTyper cannot genotype insertions, hence the missing bars in the top panels. b) Maximum F1 score for different size classes when evaluating on the presence of SVs across the whole genome. c) Size distribution of SVs in the HGSVC and GIAB catalogs." id="fig:2" style="width:90.0%" /><figcaption><span>Figure 2:</span> <strong>Structural variants from the HGSVC and Genome in a Bottle datasets</strong>. HGSVC: Simulated and real reads were used to genotype SVs and compared with the high-quality calls from Chaisson et al.<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">21</a>]</span>. Reads were simulated from the HG00514 individual. Using real reads, the three HG00514, HG00733, and NA19240 individuals were tested. GIAB: Real reads from the HG002 individual were used to genotype SVs and compared with the high-quality calls from the Genome in a Bottle consortium<span class="citation" data-cites="14neTdqfN 16GvGhO20 RCzDkVqI">[<a href="#ref-RCzDkVqI">22</a>,<a href="#ref-16GvGhO20">23</a>,<a href="#ref-14neTdqfN">25</a>]</span>. a) Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Results are separated across panels by variant type: insertions and deletions. SVTyper cannot genotype insertions, hence the missing bars in the top panels. b) Maximum F1 score for different size classes when evaluating on the presence of SVs across the whole genome. c) Size distribution of SVs in the HGSVC and GIAB catalogs.</figcaption>
</figure>
<p>We then repeated the analysis using real Illumina reads from the three HGSVC samples to benchmark the methods on a more realistic experiment. Here, vg clearly outperformed other approaches (Figures <a href="#fig:2">2</a>a and <a href="#fig:hgsvc-real-geno">S3</a>). In non-repeat regions and insertions across the whole genome, the F1 scores and precision-recall AUC were higher for vg compared to other methods. For example, for deletions in non-repeat regions, the F1 score for vg was ???? while the second best method, Paragraph, had a F1 score of ????. We observed similar results when evaluating the presence of an SV call instead of the exact genotype (Figures <a href="#fig:2">2</a>a and <a href="#fig:hgsvc-real">S4</a>). In addition, vg’s performance was stable across the spectrum of SV sizes (Figure <a href="#fig:2">2</a>b-c). Figure <a href="#fig:hgsvc-ex">3</a> shows an example of an exonic deletion that was correctly genotyped by vg but not by the other methods.</p>
<figure>
<img src="images/hgsvc-ex-del.png" alt="Figure 3: Exonic deletion in the HGSVC dataset correctly genotyped by vg. a) Visualization of the HGSVC graph as augmented by reads aligned by vg at a locus harboring a 51 bp homozygous deletion in the UTR region of the LONRF2 gene. At the bottom, a horizontal black line represents the topologically sorted nodes of the graph. Black rectangles represent edges found in the graph. Above this rendering of the topology, the reference path from GRCh38 is shown (in green). Red and blue bars represent reads mapped to the graph. Thin lines in the reference path and read mappings highlight relative gaps (either insertions or deletions) against the full graph. The vg read mappings show consistent coverage even over the deletion. b) Reads mapped to the linear genome reference GRCh38 using bwa in the same region. Reads contain soft-clipped sequences and short insertions near the deletion breakpoints. Part of the deleted region is also covered by several reads, potentially confusing traditional SV genotypers." id="fig:hgsvc-ex" style="width:80.0%" /><figcaption><span>Figure 3:</span> <strong>Exonic deletion in the HGSVC dataset correctly genotyped by vg</strong>. a) Visualization of the HGSVC graph as augmented by reads aligned by vg at a locus harboring a 51 bp homozygous deletion in the UTR region of the LONRF2 gene. At the bottom, a horizontal black line represents the topologically sorted nodes of the graph. Black rectangles represent edges found in the graph. Above this rendering of the topology, the reference path from GRCh38 is shown (in green). Red and blue bars represent reads mapped to the graph. Thin lines in the reference path and read mappings highlight relative gaps (either insertions or deletions) against the full graph. The vg read mappings show consistent coverage even over the deletion. b) Reads mapped to the linear genome reference GRCh38 using bwa in the same region. Reads contain soft-clipped sequences and short insertions near the deletion breakpoints. Part of the deleted region is also covered by several reads, potentially confusing traditional SV genotypers.</figcaption>
</figure>
<h3 id="other-long-read-datasets">Other long-read datasets</h3>
<h4 id="genome-in-a-bottle-consortium">Genome in a Bottle Consortium</h4>
<p>The Genome in a Bottle (GiaB) consortium is currently producing a high-quality SV catalog for an Ashkenazim individual (HG002)<span class="citation" data-cites="14neTdqfN 16GvGhO20 RCzDkVqI">[<a href="#ref-RCzDkVqI">22</a>,<a href="#ref-16GvGhO20">23</a>,<a href="#ref-14neTdqfN">25</a>]</span>. Dozens of SV callers operating on datasets from short, long, and linked reads were used to produce this set of SVs. We evaluated the SV genotyping methods on this sample as well using the GIAB VCF, which also contains parental calls (HG003 and HG004), all totalling 30,224 SVs. Relative to the HGSVC dataset, vg performed similarly but Paragraph saw a large boost in accuracy and was the most accurate method across all metrics. (Figures <a href="#fig:2">2</a>, <a href="#fig:giab-geno">S5</a> and <a href="#fig:giab">S6</a>, and Table <a href="#tbl:giab">S2</a>). As before, the remaining methods produced lower F1 scores.</p>
<h4 id="smrt-sv-v2-catalog-and-training-data-3nnfs6u2">SMRT-SV v2 catalog and training data <span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span></h4>
<p>A recent study by Audano et al. generated a catalog of 97,368 SVs (referred as SVPOP below) using long-read sequencing across 15 individuals<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. These variants were then genotyped from short reads across 440 individuals using the SMRT-SV v2, a machine learning-based tool implemented for that study. The SMRT-SV v2 genotyper was trained on a pseudo-diploid genome constructed from high quality assemblies of two haploid cell lines (CHM1 and CHM13) and a single negative control (NA19240). We first used vg to genotype the SVs in this two-sample training dataset using 30X coverage reads, and compared the results with the SMRT-SV v2 genotyper. vg was systematically better at predicting the presence of an SV for both SV types, but SMRT-SV v2 produced slightly better genotypes for deletions in the whole genome(see Figures <a href="#fig:chmpd-svpop">4</a>, <a href="#fig:chmpd-geno">S7</a> and <a href="#fig:chmpd">S8</a>, and Table <a href="#tbl:chmpd">S3</a>). To compare vg and SMRT-SV v2 on a largere dataset, we then genotyped SVs from the entire SVPOP catalog with both methods, using the read data from the three HGSVC samples described above. Given that the the SVPOP catalog contains these three samples, we once again evaluated accuracy by using the long-read genotypes as a baseline. Paragraph was included as an additional point of comparison.</p>
<p>Compared to SMRT-SV v2, vg had a better precision-recall curve and a higher F1 for both insertions and deletions (SVPOP in Figures <a href="#fig:chmpd-svpop">4</a> and <a href="#fig:svpop">S9</a>, and Table <a href="#tbl:svpop">S4</a>). Paragraph’s performance was virtually identical to vg’s. Of note, SMRT-SV v2 produces <em>no-calls</em> in regions where the read coverage is too low, and we observed that its recall increased when filtering these regions out the input set. Interestingly, vg performed well even in regions where SMRT-SV v2 produced <em>no-calls</em> (Figure <a href="#fig:svpop-regions">S10</a> and Table <a href="#tbl:svpop-regions">S5</a>). Audano et al. discovered 217 sequence-resolved inversions using long reads, which we attempted to genotype. vg correctly predicted the presence of around 14% of the inversions present in the three samples (Table <a href="#tbl:svpop">S4</a>). Inversions are often complex, harboring additional variation that makes their characterization and genotyping challenging.</p>
<figure>
<img src="images/chmpd-svpop-best-f1.png" alt="Figure 4: Structural variants from SMRT-SV v2 [5]. The pseudo-diploid genome built from two CHM cell lines and one negative control sample was originally used to train SMRT-SV v2 in Audano et al.[5]. It contains 16,180 SVs. The SVPOP panel shows the combined results for the HG00514, HG00733, and NA19240 individuals, three of the 15 individuals used to generate the high-quality SV catalog in Audano et al. [5]. Here, we report the maximum F1 score (y-axis) for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Genotype information is not available in the SVPOP catalog hence genotyping performance could not be evaluated." id="fig:chmpd-svpop" /><figcaption><span>Figure 4:</span> <strong>Structural variants from SMRT-SV v2 <span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span></strong>. The pseudo-diploid genome built from two CHM cell lines and one negative control sample was originally used to train SMRT-SV v2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. It contains 16,180 SVs. The SVPOP panel shows the combined results for the HG00514, HG00733, and NA19240 individuals, three of the 15 individuals used to generate the high-quality SV catalog in Audano et al. <span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. Here, we report the maximum F1 score (y-axis) for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Genotype information is not available in the SVPOP catalog hence genotyping performance could not be evaluated.</figcaption>
</figure>
<h3 id="graphs-from-alignment-of-de-novo-assemblies">Graphs from alignment of <em>de novo</em> assemblies</h3>
<p>We can construct variation graphs directly from whole genome alignments (WGA) of multiple <em>de novo</em> assemblies<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">15</a>]</span>. This bypasses the need for generating an explicit variant catalog relative to a linear reference, which could be a source of error due to the reference bias inherent in read mapping and variant calling. Genome alignments from graph-based software such as Cactus <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">26</a>]</span> can contain complex structural variation that is extremely difficult to represent, let alone call, outside of a graph, but which is nevertheless representative of the actual genomic variation between the aligned assemblies. We sought to establish if graphs built in this fashion provide advantages for SV genotyping.</p>
<p>To do so, we analyzed public sequencing datasets for 12 yeast strains from two related clades (<em>S. cerevisiae</em> and <em>S. paradoxus</em>) <span class="citation" data-cites="7f5OKa5O">[<a href="#ref-7f5OKa5O">27</a>]</span>. We distinguished two different strain sets, in order to assess how the completeness of the graph affects the results. For the <em>all strains set</em>, all 12 strains were used, with <em>S.c. S288C</em> as the reference strain. For the <em>five strains set</em>, <em>S.c. S288C</em> was used as the reference strain, and we selected two other strains from each of the two clades (see <a href="#yeast-graph-analysis">Methods</a>). We compared genotyping results from a WGA-derived graph (<em>cactus graph</em>) with results from a VCF-derived graph (<em>VCF graph</em>). The <em>VCF graph</em> was created from the linear reference genome of the <em>S.c. S288C</em> strain and a set of SVs relative to this reference strain in VCF format identified from the other assemblies in the respective strain set by three methods: Assemblytics <span class="citation" data-cites="krO7WgVi">[<a href="#ref-krO7WgVi">28</a>]</span>, AsmVar <span class="citation" data-cites="oVaXIwl5">[<a href="#ref-oVaXIwl5">29</a>]</span> and paftools <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q">30</a>]</span>. The <em>cactus graph</em> was derived from a multiple genome alignment of the strains in the respective strain set using Cactus <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">26</a>]</span>. The <em>VCF graph</em> is mostly linear and highly dependent on the reference genome. In contrast, the <em>cactus graph</em> is structurally complex and relatively free of reference bias.</p>
<p>First, we tested our hypothesis that the <em>cactus graph</em> has higher mappability due to its better representation of sequence diversity among the yeast strains (see <a href="#mappability-comparison-between-yeast-graphs">Supplementary Information</a>). Generally, more reads mapped to the <em>cactus graph</em> with high identity (Figures <a href="#fig:panel3">S13</a>a and <a href="#fig:panel5">S14</a>a) and high mapping quality (Figures <a href="#fig:panel3">S13</a>b and <a href="#fig:panel5">S14</a>b) than to the <em>VCF graph</em>. On average, 88%, 79%, and 68% of reads mapped to the <em>all strain cactus graph</em> with an identity of at least 50%, 90%, and 100%, respectively, compared to only 77%, 57%, and 23% of reads on the <em>all strain VCF graph</em>. Similarly, 88% of reads mapped to the <em>all strain cactus graph</em> with a mapping quality of at least 30 compared to only 80% of reads on the <em>all strain VCF graph</em>.</p>
<p>Next, we compared the SV genotyping performance of both graph types. We mapped short reads from the 11 non-reference strains to both graphs and genotyped variants for each strain using the vg toolkit’s variant calling module (see <a href="#toil-vg-call">Methods</a>). There is no gold standard available for these samples to compare against which renders an evaluation using recall, precision and F1 score impossible. Therefore, we used an indirect measure of SV genotyping accuracy. We evaluated each SV genotype set based on the alignment of reads to a <em>sample graph</em> constructed from the genotype set (see <a href="#calling-and-genotyping-of-svs">Methods</a>). Conceptually, the sample graph represents the sample’s diploid genome by starting out from the reference genome and augmenting it with the genotype results. If a given genotype set is correct, we expect that reads from the same sample will be mapped with high identity and confidence to the corresponding sample graph. To specifically quantify mappability in SV regions we excluded reads that produced identical mapping quality and identity on both sample graphs and an empty sample graph containing the linear reference only (see <a href="#calling-and-genotyping-of-svs">Methods</a> and Figure <a href="#fig:panel6">S15</a> for results from all reads). Then, we analyzed the average delta in mapping identity and mapping quality of the remaining short reads between both sample graphs (Figures <a href="#fig:4">5</a>a and b).</p>
<p>For most of the strains, we observed an improvement in mapping identity of the short reads on the <em>cactus sample graph</em> compared to the <em>VCF sample graph</em>. The mean improvement in mapping identity across the strains (for reads differing in mapping identity) was 8.0% and 8.5% for the <em>all strains set</em> graphs and the <em>five strains set</em> graphs, respectively. Generally, the improvement in mapping identity was larger for strains in the <em>S. paradoxus</em> clade (mean of 13.7% and 13.3% for the two strain sets, respectively) than for strains in the <em>S. cerevisiae</em> clade (mean of 3.3% and 4.4%). While the higher mapping identity indicated that the <em>cactus graph</em> represents the reads better (Figure <a href="#fig:4">5</a>a), the higher mapping quality confirmed that this did not come at the cost of added ambiguity or a more complex graph (Figure <a href="#fig:4">5</a>b). For most strains, we observed an improvement in mapping quality of the short reads on the <em>cactus sample graph</em> compared to the <em>VCF sample graph</em> (mean improvement across the strains of 1.0 and 5.7 for the two strain sets, respectively).</p>
<figure>
<img src="images/panel4.png" alt="Figure 5: SV genotyping comparison. Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in the cactus graph and the VCF graph. Subsequently, sample graphs were generated from the resulting SV genotype sets. The short reads were aligned to the sample graphs and reads with identical mapping identity and quality across both sample graphs and an additional empty sample graph were removed from the analysis. The quality of the remaining divergent alignments was used to ascertain SV genotyping performance. The bars show the average delta in mapping identity (a) and in mapping quality (b) of divergent short reads aligned to the sample graphs derived from the cactus graph and the VCF graph. Positive values denote an improvement of the cactus graph over the VCF graph. Colors represent the two strain sets and transparency indicates whether the respective strain was part of the five strains set." id="fig:4" /><figcaption><span>Figure 5:</span> <strong>SV genotyping comparison.</strong> Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in the <em>cactus graph</em> and the <em>VCF graph</em>. Subsequently, sample graphs were generated from the resulting SV genotype sets. The short reads were aligned to the sample graphs and reads with identical mapping identity and quality across both sample graphs and an additional empty sample graph were removed from the analysis. The quality of the remaining divergent alignments was used to ascertain SV genotyping performance. The bars show the average delta in mapping identity (a) and in mapping quality (b) of divergent short reads aligned to the sample graphs derived from the <em>cactus graph</em> and the <em>VCF graph</em>. Positive values denote an improvement of the <em>cactus graph</em> over the <em>VCF graph</em>. Colors represent the two strain sets and transparency indicates whether the respective strain was part of the <em>five strains set</em>.</figcaption>
</figure>
<h2 id="discussion">Discussion</h2>
<!-- Discuss why vg is doing better -->
<p>Overall, vg was the most accurate SV genotyper in our benchmarks. These results show that SV genotyping benefits from variant-aware read mapping and graph based genotyping, a finding consistent with previous studies<span class="citation" data-cites="10jxt15v0 DuODeStx 11Jy8B61m ohTIiqfV 14Uxmwbxm">[<a href="#ref-10jxt15v0">15</a>,<a href="#ref-DuODeStx">16</a>,<a href="#ref-11Jy8B61m">17</a>,<a href="#ref-ohTIiqfV">18</a>,<a href="#ref-14Uxmwbxm">19</a>]</span>. Paragraph, another graph-based genotyper which was released as we were submitting this work, was very competitive with vg and showed the best overall accuracy on the GIAB dataset. In addition to being featured prominently in Paragraph’s devleopment and evaluation, the GIAB dataset we used was a different coverage (50X) than the other 30X datasets we used. Our simulation results show that Paragraph is slightly more robust than vg with respect to differences in coverages and perhaps this is a factor in the difference in performance. In the future, we would like to better model the expected read depth in the vg genotyper as it currently does not exploit this information. In contrast, vg is much more accurate than Paragraph on the HGSVC dataset and we speculate that this is due to the higher number of overlapping variants. Using the snarl decomposition, vg can genotype arbitrary combinations of SVs simultaneously, whereas Paragraph operates one at a time.</p>
<p>We took advantage of newly released datasets for our evaluation, which feature up to 3.7 times more variants than the more widely-used GIAB benchmark. More and more large-scale projects are using low cost short-read technologies to sequence the genomes of thousands to hundreds of thousands of individuals (e.g. the Pancancer Analysis of Whole Genomes<span class="citation" data-cites="10Jid8Wql">[<a href="#ref-10Jid8Wql">31</a>]</span>, the Genomics England initiative<span class="citation" data-cites="mWj2p7Xp">[<a href="#ref-mWj2p7Xp">32</a>]</span>, and the TOPMed consortium<span class="citation" data-cites="ir1O1h8n">[<a href="#ref-ir1O1h8n">33</a>]</span>). We believe pangenome graph-based approaches will improve both how efficiently SVs can be represented, and how accurately they can be genotyped with this type of data.</p>
<!-- Input data quality: "sequence-resolved", break-point fine-tuning. -->
<p>A particular advantage of our method is that it does not require exact breakpoint resolution in the variant library. Our simulations showed that vg’s SV genotyping algorithm is robust to errors of as much as 10 bp in breakpoint location. However, there is an upper limit to this flexibility, and we find that vg cannot accurately genotype variants with much higher uncertainty in the breakpoint location (like those discovered through read coverage analysis). vg is also capable of fine-tuning SV breakpoints by augmenting the graph with differences observed in read alignments. Simulations showed that this approach can usually correct small errors in SV breakpoints (Figure <a href="#fig:simerror-bkpt">S12</a> and Table <a href="#tbl:simerror-bkpt">S6</a>).</p>
<!-- Already superior but will only get better with new vg dev -->
<p>vg uses a unified framework to call and score different variant types simultaneously. In this work, we only considered graphs containing certain types of SVs, but the same methods can be extended to a broader range of graphs. For example, we are interested in evaluating how genotyping SVs together with SNPs and small indels using a combined graph effects the accuracy of studying either alone. The same methods used for genotyping known variants in this work can also be extended to call novel variants by first augmenting the graph with edits from the mapped reads. This approach, which was used only in the breakpoint fine-tuning portion of this work, could be further used to study small variants around and nested within SVs. Novel SVs could be called by augmenting the graph with long-read mappings. vg is entirely open source, and its ongoing development is supported by a growing community of researchers and users with common interest in scalable, unbiased pangenomic analyses and representation. We expect this collaboration to continue to foster increases in the speed, accuracy and applicability of methods based on pangenome graphs in the years ahead.</p>
<!-- Benefits of *de novo* assemblies -->
<p>Our results suggest that constructing a graph from <em>de novo</em> assembly alignment instead of a VCF leads to better SV genotyping. High quality <em>de novo</em> assemblies for human are becoming more and more common due to improvements in technologies like optimized mate-pair libraries<span class="citation" data-cites="pJAv1D8R">[<a href="#ref-pJAv1D8R">34</a>]</span> and long-read sequencing<span class="citation" data-cites="6KbgcueR">[<a href="#ref-6KbgcueR">12</a>]</span>. We expect future graphs to be built from the alignment of numerous <em>de novo</em> assemblies, and we are presently working on scaling our assembly-based pipeline to human-sized genome assemblies. Another challenge is creating genome graphs that integrate assemblies with variant-based data resources. One possible approach is to progressively align assembled contigs into variation graphs constructed from variant libraries, but methods for doing so are still experimental.</p>
<h2 id="conclusion">Conclusion</h2>
<p>In this study, the vg toolkit was compared to existing SV genotypers across several high-quality SV catalogs. We showed that its method of mapping reads to a variation graph leads to better SV genotyping compared to other state of the art methods. This work introduces a flexible strategy to integrate the growing number of SVs being discovered with higher resolution technologies into a unified framework for genome inference. Our work on whole genome alignment graphs shows the benefit of directly utilizing <em>de novo</em> assemblies rather than variant catalogs to integrate SVs in genome graphs. We expect this latter approach to increase in significance as the reduction in long read sequencing costs drives the creation of numerous new <em>de novo</em> assemblies. We envision a future in which the lines between variant calling, genotyping, alignment, and assembly are blurred by rapid changes in sequencing technology. Fully graph based approaches, like the one we present here, will be of great utility in this new phase of genome inference.</p>
<h2 id="methods">Methods</h2>
<h3 id="sv-genotyping-algorithm">SV Genotyping Algorithm</h3>
<p>The input to the SV genotyping algorithm is an indexed variation graph in <code>xg</code> format along with a (single-sample) read alignment in <code>GAM</code> format. If the graph was constructed from a VCF, as was the case for the human-genome graphs discussed in this paper, this VCF can also be input to the caller. The first step is to compute a compressed coverage index from the alignment using this command, <code>vg pack &lt;graph.xg&gt; &lt;alignment.gam&gt; -Q 5 -o graph.pack</code>. This index stores the number of reads with mapping quality at least 5 mapped to each edge and each base of each node on the graph. Computing the coverage can be done in a single scan through the reads and, in practice, tends to be an order of magnitude faster than sorting the reads.</p>
<p>Variation graphs, as represented in vg, are bidirected. In a bidirected graph, every node can be thought of having two distinct <em>sides</em>. See, for example, the left and right sides of each rectangle in Figure {<span class="citation" data-cites="fig:vg-sv-cartoon">[<span class="citeproc-not-found" data-reference-id="fig:vg-sv-cartoon"><strong>???</strong></span>]</span>}. If <em>x</em> is the side of a given node <em>A</em>, then we use the notation <em>x’</em> to denote the other side of <em>A</em>. A snarl is defined by a pair of sides, <em>x</em> and <em>y</em>, that satisfy the following criteria: 1. Removing all edges incident to <em>x’</em> and <em>y’</em> disconnects the graph, creating a connected component <em>X</em> that contains <em>x</em> and <em>y</em>. 2. There is no side <em>z</em> in <em>X</em> such that <em>{x,z}</em> satisfies the above criteria. Likewise for <em>y</em>.</p>
<p>Snarls can be computed in linear time using a cactus graph decomposition <span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">24</a>]</span>. They can be computed once for a given graph using <code>vg snarls</code>, or on the fly with <code>vg call</code>.</p>
<p>Once the snarls have been identified, the SV genotyping algorithm proceeds as follows. For every snarl in the graph such that both its end nodes lie on a reference path (such as a chromosome) and that it is not contained in another snarl, the following steps are performed.</p>
<ol type="1">
<li>All VCF variants, <em>v1, v2, …, vk</em> that are contained within the snarl are looked up using information embedded during graph construction. Let <em>|vi|</em> be the number of alleles in the <em>ith</em> VCF variant. Then there are <em>|v1|</em>x<em>|v2|</em>…x<em>|vk|</em> possible haplotypes through the snarl. If this number is too high (&gt;500k), then alleles with average support of less than 1 are filtered out.</li>
<li>For each possible haplotype, a bidrected path through the snarl (from <em>x</em> to <em>y</em>) is computed that corresponds to it. Annotation information from the VCF that was embedded in the graph during construction is used for this.</li>
<li>For each haplotype path, its average support (over bases and edges) is computed using the compressed coverage index, and the two most-supported paths are selected (ties are broken arbitrarily).</li>
<li>If the most supported path exceeds the minimum support threshold (default 1), and has more than <em>B</em> (default 6) times the support of the next most supported path, the site is called homozygous for the allele associated with the most supported path.</li>
<li>Else if the second most supported path exceeds the minimum support threshold (default 1), then the site is deemed heterozygous with an allele from each of the top two paths.</li>
<li>Given the genotype computed above, it is trivial to map back from the chosen paths to the VCF alleles in order to produce the final output</li>
</ol>
<p>The command to do the above is <code>vg call &lt;graph.xg&gt; -k &lt;graph.pack&gt; -v variants.vcf.gz</code> If the graph was not constructed from a VCF, then a similar algorithm is used except the traversals are computed heuristically searching through the graph. This is enabled by not using the <code>-v</code> option in the above command.</p>
<h3 id="toil-vg">toil-vg</h3>
<p>toil-vg is a set of Python scripts for simplifying vg tasks such as graph construction, read mapping and SV genotyping. Much of the analysis in this report was done using toil-vg, with the exact commands available at <a href="https://github.com/vgteam/sv-genotyping-paper">github.com/vgteam/sv-genotyping-paper</a>. toil-vg uses the Toil workflow engine <span class="citation" data-cites="faeC2cx0">[<a href="#ref-faeC2cx0">35</a>]</span> to seamlessly run pipelines locally, on clusters or on the cloud. Graph indexing, and mapping in particular are computationally expensive (though work is underway to address this) and well-suited to distribution on the cloid. The principal toil-vg commands used are described below.</p>
<h4 id="toil-vg-construct">toil-vg construct</h4>
<p>toil-vg construct automates graph construction and indexing following the best practices put forth by the vg community. Graph construction is parallelized across different sequences from the reference FASTA, and different whole-genome indexes are created side by side when possible. The graph is automatically annotated with paths corresponding to the different alleles in the input VCF. The indexes created are the following:</p>
<ul>
<li>xg index: This is a compressed version of the graph that allows fast node, edge and path lookups</li>
<li>gcsa2 index: This is a substring index used only for read mapping</li>
<li>gbwt index: This is an index of all the haplotypes in the VCF as implied by phasing infromation. When available, it is used to help ensure that haplotype information is preserved when constructing the gcsa2 index</li>
<li>snarls index: The snarls represent sites of varition in the graph and are used for genotyping and variant calling.</li>
</ul>
<h4 id="toil-vg-map">toil-vg map</h4>
<p>toil-vg map splits the input reads into batches, maps each batch in parallel, then merges the result.</p>
<h4 id="toil-vg-call">toil-vg call</h4>
<p>toil-vg call splits the input graph by chromosome and calls each one individually. <code>vg call</code> has been recently updated so that this subdivision is largely unecessary: the entire graph can be easily called at once. Still, toil-vg can be used to farm this task out to a single cloud node if desired.</p>
<h4 id="toil-vg-sveval">toil-vg sveval</h4>
<p>toil-vg sveval evaluates the SV calls relative to a truth set. The variants are first normalized with <code>bcftools norm</code> (1.9) to ensure consistent representation between called variants and baseline variants<span class="citation" data-cites="LAG2q9WK">[<a href="#ref-LAG2q9WK">36</a>]</span>. We then implemented an overlap-based strategy to compare SVs and compute evaluation metrics (sveval R package: <a href="https://github.com/jmonlong/sveval">https://github.com/jmonlong/sveval</a>). Figure <a href="#fig:sveval">S16</a> shows an overview of the SV evaluation approach which is described below.</p>
<p>For deletions and inversions, we begin by computing the overlaps between the SVs in the call set and the truth set. For each variant we then compute the proportion of its region that is covered by a variant in the other set, considering only variants overlapping with at least 10% reciprocal overlap. If this coverage proportion is higher than 50%, we consider the variant <em>covered</em>. True positives (TPs) are covered variants from the call set (when computing the precision) or the truth set (when computing the recall). Variants from the call set are considered false positives (FPs) if they are not covered by the truth set. Conversely, variants from the truth set are considered false negatives (FNs) if they are not covered by the call set.</p>
<p>For insertions, we select pairs of insertions that are located no farther than 20 bp from each other. We then align the inserted sequences using a Smith-Waterman alignment. For each insertion we compute the proportion of its inserted sequence that aligns a matched variant in the other set. If this proportion is at least 50% the insertions are considered covered. Covering relationships are used to define TPs, FPs, and FNs the same way as for deletions and inversions.</p>
<p>The results shown in this study used a minimum of 50% coverage to match variants but we also replicated the results using 90% minimum coverage and observed similar results (see Figure <a href="#fig:eval-stringent">S11</a>).</p>
<p>The coverage statistics are computed using any variant larger than 1 bp but a minimum size is required for a variant to be counted as TP, FP, or FN. In this work, we used the default minimum SV size of 50 bp.</p>
<p>sveval accepts VCF files with symbolic or explicit representation of the SVs. If the explicit representation is used, multi-allelic variants are split and their sequences right-trimmed. When using the explicit representation and when the REF and ALT sequences are longer than 10 bp, the reverse-complement of the ALT sequence is aligned to the REF sequence to identify potential inversions. If more than 80% of the sequence aligns, it is classified as an inversion.</p>
<p>We assess both the ability to predict the presence of an SV as well as the full genotype. For the <em>presence</em> evaluation, both heterozygous and homozygous alternate SVs are compared jointly using the approach described above. To compute genotype-level metrics, the heterozygous and homozygous SVs are compared separately. Before splitting the variants by genotype, pairs of heterozygous variants with reciprocal overlap of at least 80% are merged into a homozygous ALT variant. To handle fragmented variants, consecutive heterozygous variants located at less that 20 bp from each other are first merged into larger heterozygous variants.</p>
<p>Precision-recall curves are produced by successively filtering out variants of low-quality. By default the <em>QUAL</em> field in the VCF file is used as the quality information. If <em>QUAL</em> is missing (or contains only 0s), the genotype quality in the <em>GQ</em> field is used.</p>
<p>The evaluation is performed using all variants or using only variants within high-confidence regions. In most analysis, the high-confidence regions are constructed by excluding segmental duplications and tandem repeats (using the respective tracks from the UCSC Genome Browser). For the GIAB analysis, we used the Tier 1 high-confidence regions provided by the GIAB consortium in version 0.6.</p>
<h3 id="other-sv-genotypers">Other SV genotypers</h3>
<h4 id="bayestyper-v1.5-beta-62888d6">BayesTyper (v1.5 beta 62888d6)</h4>
<p>Where not specified otherwise BayesTyper was run as follows. Raw reads were mapped to the reference genome using bwa mem (citation: https://arxiv.org/abs/1303.3997) (0.7.17). GATK haplotypecaller<span class="citation" data-cites="NCr4QkOg">[<a href="#ref-NCr4QkOg">37</a>]</span> (3.8) and Platypus<span class="citation" data-cites="1DYmkalz4">[<a href="#ref-1DYmkalz4">38</a>]</span> (0.8.1.1) with assembly enabled were run on the mapped reads to call SNVs and short indels (&lt;50bp) needed by BayesTyper for correct genotyping. The VCFs with these variants were then normalised using <code>bcftools norm</code> (1.9) and combined with the SVs across samples using <code>bayesTyperTools combine</code> to produce the input candidate set. k-mers in the raw reads were counted using kmc (citation: https://academic.oup.com/bioinformatics/article/33/17/2759/3796399) (3.1.1) with a k-mer size of 55. A Bloom filter was constructed from these k-mers using <code>bayesTyperTools makeBloom</code>. Finally, variants were clustered and genotyped using <code>bayestyper cluster</code> and <code>bayestyper genotype</code>, respectively, with default parameters except <code>--min-genotype-posterior 0</code>. Non-PASS variants and non-SVs (GATK and Platypus origin) were filtered prior to evaluation using <code>bcftools filter</code> and <code>filterAlleleCallsetOrigin</code>, respectively.</p>
<h4 id="delly-v0.7.9">Delly (v0.7.9)</h4>
<p>The <code>delly call</code> command was run on the reads mapped by <code>bwa mem</code>, the reference genome FASTA file, and the VCF containing the SVs to genotype (converted to their explicit representations).</p>
<h4 id="svtyper-v0.7.0">SVTyper (v0.7.0)</h4>
<p>The VCF containing deletions was converted to symbolic representation and passed to <code>svtyper</code> with the reads mapped by <code>bwa mem</code>. The output VCF was converted back to explicit representation using <code>bayesTyperTools convertAllele</code> to facilitate variant normalization before evaluation.</p>
<h4 id="paragraph-v2.3">Paragraph (v2.3)</h4>
<p>Paragraph was run using default parameters using the <code>multigrmpy.py</code> script, taking the input VCF and reads mapped by <code>bwa mem</code> as inputs. We used the genotype estimates in the <code>genotypes.vcf.gz</code> output file. In order for Paragraph to run, we added padding sequence to problematic variants in the input VCFs of the GIAB and SVPOP catalogs.</p>
<h4 id="smrt-sv-v2-v2.0.0-feb-21-2019-commit-adb13f2">SMRT-SV v2 (v2.0.0 Feb 21 2019 commit adb13f2)</h4>
<p>SMRT-SV v2 was run with the “30x-4” model and min-call-depth 8 cutoff. It was run only on VCFs created by SMRT-SV, for which the required contig BAMs were available. The Illumina BAMs used where the same as the other methods described above. The output VCF was converted back to explicit representation to facilitate variant normalization later.</p>
<p>Running times for the different tools are shown in Table <a href="#tbl:timing">S7</a>.</p>
<h3 id="simulation-experiment">Simulation experiment</h3>
<p>We simulated a synthetic genome with 1000 insertions, deletions and inversions. We separated each variant from the next by a buffer of at least 500 bp. The sizes of deletions and insertions followed the distribution of SV sizes from the HGSVC catalog. We used the same size distribution as deletions for inversions. A VCF file was produced for three simulated samples with genotypes chosen uniformly between homozygous reference, heterozygous, and homozygous alternate.</p>
<p>We created another VCF file containing errors in the SV breakpoint locations. We shifted one or both breakpoints of deletions and inversions by distances between 1 and 10 bp. The locations and sequences of insertions were also modified, either shifting the variants or shortening them at the flanks, again by up to 10 bp.</p>
<p>Paired-end reads were simulated using <code>vg sim</code> on the graph that contained the true SVs. Different read depths were tested: 1x, 3x, 7x, 10x, 13x, 20x. The base qualities and sequencing errors were trained to resemble real Illumina reads from NA12878 provided by the Genome in a Bottle Consortium.</p>
<p>The genotypes called in each experiment (genotyping method/VCF with or without errors/sequencing depth) were compared to the true SV genotypes to compute the precision, recall and F1 score (see <a href="#toil-vg-sveval">toil-vg sveval</a>).</p>
<h4 id="breakpoint-fine-tuning-using-graph-augmentation">Breakpoint fine-tuning using graph augmentation</h4>
<p>vg can call variants after augmenting the graph with the read alignments to discover new variants (see <a href="#toil-vg-call">toil-vg call</a>). We tested if this approach could fine-tune the breakpoint location of SVs in the graph. We started with the graph that contained approximate SVs (1-10 bp errors in breakpoint location) and 20x simulated reads from the simulation experiment (see <a href="#simulation-experiment">Simulation experiment</a>). The variants called after graph augmentation were compared with the true SVs. We considered fine-tuning correct if the breakpoints matched exactly.</p>
<h3 id="hgsvc-analysis">HGSVC Analysis</h3>
<p>We first obtained phased VCFs for the three Human Genome Structural Variation Consortium (HGSVC) samples from Chaisson et al.<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">21</a>]</span> and combined them with <code>bcftools merge</code>. A variation graph was created and indexed using the combined VCF and the HS38D1 reference with alt loci excluded. The phasing information was used to construct a GBWT index<span class="citation" data-cites="jj4JJlsg">[<a href="#ref-jj4JJlsg">39</a>]</span>, from which the two haploid sequences from HG00514 were extracted as a graph. Illumina read pairs with 30x coverage were simulated from these sequences using vg sim, with an error model learned from real reads from the same sample. These simulated reads reflect an idealized situation where the breakpoints of the SVs being genotyped are exactly known <em>a priori</em>. The reads were mapped to the graph, and the mappings used to genotype the SVs in the graph. Finally, the SV calls were compared back to the HG00514 genotypes from the HGSVC VCF. We repeated the process with the same reads on the linear reference, using <code>bwa mem</code> for mapping and Delly, SVTyper and BayesTyper for SV genotyping.</p>
<p>We downloaded Illumina HiSeq 2500 paired end reads from the EBI’s ENA FTP site for the three samples, using Run Accessions ERR903030, ERR895347 and ERR894724 for HG00514, HG00733 and NA19240, respectively. We ran the graph and linear mapping and genotyping pipelines exactly as for the simulation, and aggregated the comparison results across the three samples. We used BayesTyper to jointly genotype the 3 samples.</p>
<h3 id="giab-analysis">GIAB Analysis</h3>
<p>We obtained version 0.5 of the Genome in a Bottle (GIAB) SV VCF for the Ashkenazim son (HG002) and his parents from the NCBI FTP site. We obtained Illumina reads as described in Garrison et al.<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">15</a>]</span> and downsampled them to 50x coverage. We used these reads as input for <code>vg call</code> and the other SV genotyping pipelines described above (though with GRCh37 instead of GRCh38). For BayesTyper, we created the input variant set by combining the GIAB SVs with SNV and indels from the same study. Variants with reference allele or without a determined genotype for HG002 in the GIAB call set (10,569 out of 30,224) were considered “false positives” as a proxy measure for precision. These variants correspond to putative technical artifacts and parental calls not present in HG002. For the evaluation in high confidence regions, we used the Tier 1 high-confidence regions provided by the GIAB consortium in version 0.6.</p>
<h3 id="smrt-sv-v2-comparison-chmpd-and-svpop">SMRT-SV v2 Comparison (CHMPD and SVPOP)</h3>
<p>The SMRT-SV v2 genotyper can only be used to genotype sequence-resolved SVs present on contigs with known SV breakpoints, such as those created by SMRT-SV v2, and therefore could not be run on the simulated, HGSVC, or GIAB call sets. The authors shared their training and evaluation set: a pseudodiploid sample constructed from combining the haploid CHM1 and CHM13 samples (CHMPD), and a negative control (NA19240). The high quality of the CHM assemblies makes this set an attractive alternative to using simulated reads. We used this two-sample pseudodiploid VCF along with the 30X read set to construct, map and genotype with vg, and also ran SMRT-SV v2 genotyper with the “30x-4” model and min-call-depth 8 cutoff, and compared the two back to the original VCF.</p>
<p>In an effort to extend this comparison from the training data to a more realistic setting, we reran the three HGSVC samples against the SMRT-SV v2 discovery VCF (SVPOP, which contains 12 additional samples in addition to the three from HGSVC) published by Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span> using vg and SMRT-SV v2 Genotyper. The discovery VCF does not contain genotypes. In consequence, we were unable to distinguish between heterozygous and homozygous genotypes, and instead considered only the presence or absence of a non-reference allele for each variant.</p>
<p>SMRT-SV v2 produces explicit <em>no-call</em> predictions when the read coverage is too low to produce accurate genotypes. These no-calls are considered homozygous reference in the main accuracy evaluation. We also explored the performance of vg and SMRT-SV v2 in different sets of regions (Figure <a href="#fig:svpop-regions">S10</a> and Table <a href="#tbl:svpop-regions">S5</a>):</p>
<ol type="1">
<li>Non-repeat regions, i.e. excluding segmental duplications and tandem repeats (using the respective tracks from the UCSC Genome Browser).</li>
<li>Repeat regions defined as segmental duplications and tandem repeats.</li>
<li>Regions where SMRT-SV v2 could call variants.</li>
<li>Regions where SMRT-SV v2 produced no-calls.</li>
</ol>
<h3 id="yeast-graph-analysis">Yeast graph analysis</h3>
<p>For the analysis of graphs from <em>de novo</em> assemblies, we utilized publicly available PacBio-derived assemblies and Illumina short read sequencing datasets for 12 yeast strains from two related clades (Table <a href="#tbl:strains">1</a>) <span class="citation" data-cites="7f5OKa5O">[<a href="#ref-7f5OKa5O">27</a>]</span>. We constructed graphs from two different strain sets: For the <em>five strains set</em>, we selected five strains for graph contruction (<em>S.c. SK1</em>, <em>S.c. YPS128</em>, <em>S.p. CBS432</em>, <em>S.p. UFRJ50816</em> and <em>S.c. S288C</em>). We randomly selected two strains from different subclades of each clade as well as the reference strain <em>S.c. S288C</em>. For the <em>all strains set</em> in contrast, we utilized all twelve strains for graph contruction. We constructed two different types of genome graphs from the PacBio-derived assemblies of the five or twelve (depending on the strains set) selected strains. In this section, we describe the steps for the construction of both graphs and the genotyping of variants. More details and the precise commands used in our analyses can be found at <a href="https://github.com/vgteam/sv-genotyping-paper">github.com/vgteam/sv-genotyping-paper</a>.</p>
<a name="tbl:strains"></a>
<table>
<caption><span>Table 1:</span> 12 yeast strains from two related clades were used in our analysis. Five strains were selected to be included in the <em>five strains set</em> and all strains were included in the <em>all strains set</em>. Graphs were constructed from strains in the respective strain set while all eleven non-reference strains were used for genotyping. </caption>
<thead>
<tr class="header">
<th>Strain</th>
<th>Clade</th>
<th>Included in <em>five strains set</em></th>
<th>Included in <em>all strains set</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>S288C</td>
<td>S. cerevisiae</td>
<td>✓</td>
<td>✓</td>
</tr>
<tr class="even">
<td>SK1</td>
<td>S. cerevisiae</td>
<td>✓</td>
<td>✓</td>
</tr>
<tr class="odd">
<td>YPS128</td>
<td>S. cerevisiae</td>
<td>✓</td>
<td>✓</td>
</tr>
<tr class="even">
<td>UWOPS034614</td>
<td>S. cerevisiae</td>
<td></td>
<td>✓</td>
</tr>
<tr class="odd">
<td>Y12</td>
<td>S. cerevisiae</td>
<td></td>
<td>✓</td>
</tr>
<tr class="even">
<td>DBVPG6765</td>
<td>S. cerevisiae</td>
<td></td>
<td>✓</td>
</tr>
<tr class="odd">
<td>DBVPG6044</td>
<td>S. cerevisiae</td>
<td></td>
<td>✓</td>
</tr>
<tr class="even">
<td>CBS432</td>
<td>S. paradoxus</td>
<td>✓</td>
<td>✓</td>
</tr>
<tr class="odd">
<td>UFRJ50816</td>
<td>S. paradoxus</td>
<td>✓</td>
<td>✓</td>
</tr>
<tr class="even">
<td>N44</td>
<td>S. paradoxus</td>
<td></td>
<td>✓</td>
</tr>
<tr class="odd">
<td>UWOPS919171</td>
<td>S. paradoxus</td>
<td></td>
<td>✓</td>
</tr>
<tr class="even">
<td>YPS138</td>
<td>S. paradoxus</td>
<td></td>
<td>✓</td>
</tr>
</tbody>
</table>
<h4 id="construction-of-the-vcf-graph">Construction of the <em>VCF graph</em></h4>
<p>We constructed the first graph (called the <em>VCF graph</em> throughout the paper) by adding variants onto a linear reference. This method requires one assembly to serve as a reference genome. The other assemblies must be converted to variant calls relative to this reference. The PacBio assembly of the S.c. S288C strain was chosen as the reference genome because this strain was used for the S. cerevisiae genome reference assembly. To obtain variants for the other assemblies, we combined three methods for SV detection from genome assemblies: Assemblytics <span class="citation" data-cites="krO7WgVi">[<a href="#ref-krO7WgVi">28</a>]</span> (commit df5361f), AsmVar (commit 5abd91a) <span class="citation" data-cites="oVaXIwl5">[<a href="#ref-oVaXIwl5">29</a>]</span> and paftools (version 2.14-r883) <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q">30</a>]</span>. We constructed a union set of SVs detected by the three methods (using bedtools <span class="citation" data-cites="1HWiAHnIw">[<a href="#ref-1HWiAHnIw">40</a>]</span>), and combined variants with a reciprocal overlap of at least 50% to avoid duplication in the union set. We merged these union sets of variants for each of the other (non-reference) strains in the strain set, and we then applied another deduplication step to combine variants with a reciprocal overlap of at least 90%. We then used <code>vg construct</code> to build the <em>VCF graph</em> with the total set of variants and the linear reference genome.</p>
<h4 id="construction-of-the-cactus-graph">Construction of the <em>cactus graph</em></h4>
<p>The second graph (called the <em>cactus graph</em> throughout the paper) was constructed from a whole genome alignment between the assemblies. First, the repeat-masked PacBio-assemblies of the strains in the strain set were aligned with our Cactus tool <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">26</a>]</span>. Cactus requires a phylogenetic tree of the strains which was estimated using Mash (version 2.1) <span class="citation" data-cites="mH9pzoIn">[<a href="#ref-mH9pzoIn">41</a>]</span> and PHYLIP (version 3.695) <span class="citation" data-cites="tIvRXd6o">[<a href="#ref-tIvRXd6o">42</a>]</span>. Subsequently, we converted the HAL format output file to a variation graph with hal2vg (<a href="https://github.com/ComparativeGenomicsToolkit/hal2vg">https://github.com/ComparativeGenomicsToolkit/hal2vg</a>).</p>
<h4 id="genotyping-of-svs">Genotyping of SVs</h4>
<p>Prior to genotyping, we mapped the Illumina short reads of all 12 yeast strains to both graphs using <code>vg map</code>. We measured the fractions of reads mapped with specific properties using <code>vg view</code> and the JSON processor <code>jq</code>. Then, we applied <code>toil-vg call</code> (commit be8b6da) to genotype variants, obtaining a separate genotype set for each of the 11 non-reference strains on both graphs and for each of the two strain sets (in total 11 x 2 x 2 = 44 genotype sets). From the genotype sets, we removed variants smaller than 50 bp and variants with missing or homozygous reference genotypes. To evaluate the filtered genotype sets, we generated a sample graph (i.e. a graph representation of the genotype set) for each genotype set using <code>vg construct</code> and <code>vg mod</code> on the reference assembly <em>S.c. S288C</em> and the genotype set. Subsequently, we mapped short reads from the respective strains to each sample graph using <code>vg map</code>. We mapped the short reads also to an empty sample graph that was generated using <code>vg construct</code> as a graph representation of the linear reference genome. In an effort to restrict our analysis to SV regions, we removed reads that mapped equally well (i.e. with identical mapping quality and percent identity) to all three graphs (the two sample graphs and the empty sample graph) from the analysis. These filtered out reads most likely stem from portions of the strains’ genomes that are identical to the reference strain <em>S.c. S288C</em>. We analyzed the remaining alignments of reads from SV regions with <code>vg view</code> and <code>jq</code>.</p>
<h2 id="declarations">Declarations</h2>
<h3 id="availability-of-data-and-material">Availability of data and material</h3>
<p>The commands used to run the analyses presented in this study are available at <a href="https://github.com/vgteam/sv-genotyping-paper">github.com/vgteam/sv-genotyping-paper</a><span class="citation" data-cites="rXSPQWTb">[<a href="#ref-rXSPQWTb">43</a>]</span>. The datasets generated and/or analysed during the current study are also listed in this repository.</p>
<p>The scripts to generate the manuscript, including figures and tables, are available at <a href="https://github.com/jmonlong/manu-vgsv">github.com/jmonlong/manu-vgsv</a>.</p>
<h3 id="competing-interests">Competing interests</h3>
<p>The authors declare that they have no competing interests.</p>
<h3 id="funding">Funding</h3>
<p>Research reported in this publication was supported by the National Human Genome Research Institute of the National Institutes of Health under Award Number U54HG007990 and U01HL137183. This publication was supported by a Subagreement from European Molecular Biology Laboratory with funds provided by Agreement No. 2U41HG007234 from National Institute of Health, NHGRI. Its contents are solely the responsibility of the authors and do not necessarily represent the official views of National Institute of Health, NHGRI or European Molecular Biology Laboratory. The research was made possible by the generous financial support of the W.M. Keck Foundation (DT06172015).</p>
<p>DH was supported by the International Max Planck Research School for Computational Biology and Scientific Computing doctoral program. JE was supported by the Jack Baskin and Peggy Downes-Baskin Fellowship. AMN was supported by the National Institutes of Health (5U41HG007234), the W.M. Keck Foundation (DT06172015) and the Simons Foundation (SFLIFE# 35190). JAS was supported by the Carlsberg Foundation.</p>
<h3 id="authors-contributions">Authors’ contributions</h3>
<p>EG, AN, GH, JS, JE and ED implemented the read mapping and variant calling in the vg toolkit. GH, DH, JM, JAS and EG performed analysis on the different datasets. GH, DH, JM and BP designed the study. GH, DH and JM drafted the manuscript. All authors read, reviewed, and approved the final manuscript.</p>
<h3 id="acknowledgements">Acknowledgements</h3>
<p>We thank Peter Audano for sharing the CHMPD dataset and for his assistance with SMRT-SV v2.</p>
<h3 id="authors-information">Authors’ information</h3>
<p>These authors contributed equally: Glenn Hickey, David Heller, Jean Monlong.</p>
<h2 id="supplementary-material" class="page_break_before">Supplementary Material</h2>
<h3 id="supplementary-tables">Supplementary Tables</h3>
<a name="tbl:hgsvc"></a>
<table>
<caption><span>Table S1:</span> Genotyping evaluation on the HGSVC dataset. Precision, recall and F1 score for the call set with the best F1 score. The best F1 scores were achieved with no filtering in the vast majority of cases (see Figure <a href="#fig:hgsvc-sim-geno">S1</a> and <a href="#fig:hgsvc-real-geno">S3</a>). The numbers in parentheses corresponds to the results in non-repeat regions. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Experiment</th>
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Type</th>
<th style="text-align: left;">Precision</th>
<th style="text-align: left;">Recall</th>
<th style="text-align: left;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Simulated reads</td>
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.795 (0.885)</td>
<td style="text-align: left;">0.796 (0.883)</td>
<td style="text-align: left;">0.795 (0.884)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.869 (0.971)</td>
<td style="text-align: left;">0.771 (0.92)</td>
<td style="text-align: left;">0.817 (0.945)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.91 (0.935)</td>
<td style="text-align: left;">0.835 (0.9)</td>
<td style="text-align: left;">0.871 (0.917)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.898 (0.981)</td>
<td style="text-align: left;">0.806 (0.929)</td>
<td style="text-align: left;">0.849 (0.954)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">SVTyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.809 (0.876)</td>
<td style="text-align: left;">0.328 (0.754)</td>
<td style="text-align: left;">0.467 (0.81)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.767 (0.866)</td>
<td style="text-align: left;">0.093 (0.225)</td>
<td style="text-align: left;">0.166 (0.358)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.696 (0.903)</td>
<td style="text-align: left;">0.707 (0.846)</td>
<td style="text-align: left;">0.701 (0.874)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Real reads</td>
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.431 (0.683)</td>
<td style="text-align: left;">0.541 (0.726)</td>
<td style="text-align: left;">0.48 (0.704)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.65 (0.886)</td>
<td style="text-align: left;">0.519 (0.708)</td>
<td style="text-align: left;">0.577 (0.787)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.601 (0.747)</td>
<td style="text-align: left;">0.254 (0.433)</td>
<td style="text-align: left;">0.357 (0.549)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.627 (0.91)</td>
<td style="text-align: left;">0.325 (0.381)</td>
<td style="text-align: left;">0.428 (0.537)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">SVTyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.661 (0.733)</td>
<td style="text-align: left;">0.236 (0.551)</td>
<td style="text-align: left;">0.348 (0.629)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.516 (0.621)</td>
<td style="text-align: left;">0.068 (0.176)</td>
<td style="text-align: left;">0.12 (0.275)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.55 (0.838)</td>
<td style="text-align: left;">0.445 (0.547)</td>
<td style="text-align: left;">0.492 (0.662)</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:giab"></a>
<table>
<caption><span>Table S2:</span> Genotyping evaluation on the Genome in a Bottle dataset. Precision, recall and F1 score for the call set with the best F1 score. The best F1 scores were achieved with no filtering in the vast majority of cases (see Figure <a href="#fig:giab-geno">S5</a>). The numbers in parentheses corresponds to the results in non-repeat regions. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Type</th>
<th style="text-align: left;">Precision</th>
<th style="text-align: left;">Recall</th>
<th style="text-align: left;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.658 (0.774)</td>
<td style="text-align: left;">0.646 (0.735)</td>
<td style="text-align: left;">0.652 (0.754)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.68 (0.768)</td>
<td style="text-align: left;">0.643 (0.735)</td>
<td style="text-align: left;">0.661 (0.751)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.776 (0.879)</td>
<td style="text-align: left;">0.286 (0.379)</td>
<td style="text-align: left;">0.418 (0.53)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.808 (0.886)</td>
<td style="text-align: left;">0.512 (0.696)</td>
<td style="text-align: left;">0.627 (0.779)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SVTyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.742 (0.818)</td>
<td style="text-align: left;">0.342 (0.496)</td>
<td style="text-align: left;">0.468 (0.618)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.822 (0.894)</td>
<td style="text-align: left;">0.177 (0.268)</td>
<td style="text-align: left;">0.291 (0.412)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.722 (0.822)</td>
<td style="text-align: left;">0.645 (0.768)</td>
<td style="text-align: left;">0.681 (0.794)</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:chmpd"></a>
<table>
<caption><span>Table S3:</span> Genotyping evaluation on the pseudo-diploid genome built from CHM cell lines in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.665</td>
<td style="text-align: right;">0.661</td>
<td style="text-align: right;">0.663</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.688</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.579</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.806</td>
<td style="text-align: right;">0.784</td>
<td style="text-align: right;">0.795</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.869</td>
<td style="text-align: right;">0.762</td>
<td style="text-align: right;">0.812</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.757</td>
<td style="text-align: right;">0.536</td>
<td style="text-align: right;">0.628</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.848</td>
<td style="text-align: right;">0.630</td>
<td style="text-align: right;">0.723</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.880</td>
<td style="text-align: right;">0.680</td>
<td style="text-align: right;">0.767</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.971</td>
<td style="text-align: right;">0.824</td>
<td style="text-align: right;">0.891</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:svpop"></a>
<table>
<caption><span>Table S4:</span> Calling evaluation on the SVPOP dataset. Combined results for the HG00514, HG00733 and NA19240 individuals, 3 of the 15 individuals used to generate the high-quality SV catalog in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">TP</th>
<th style="text-align: right;">FP</th>
<th style="text-align: right;">FN</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">25838</td>
<td style="text-align: right;">22042</td>
<td style="text-align: right;">15772</td>
<td style="text-align: right;">0.540</td>
<td style="text-align: right;">0.621</td>
<td style="text-align: right;">0.577</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">14545</td>
<td style="text-align: right;">6824</td>
<td style="text-align: right;">15425</td>
<td style="text-align: right;">0.681</td>
<td style="text-align: right;">0.485</td>
<td style="text-align: right;">0.567</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">173</td>
<td style="text-align: right;">0.509</td>
<td style="text-align: right;">0.135</td>
<td style="text-align: right;">0.213</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">8051</td>
<td style="text-align: right;">3258</td>
<td style="text-align: right;">1817</td>
<td style="text-align: right;">0.712</td>
<td style="text-align: right;">0.816</td>
<td style="text-align: right;">0.760</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3769</td>
<td style="text-align: right;">623</td>
<td style="text-align: right;">818</td>
<td style="text-align: right;">0.858</td>
<td style="text-align: right;">0.822</td>
<td style="text-align: right;">0.840</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">75</td>
<td style="text-align: right;">0.613</td>
<td style="text-align: right;">0.202</td>
<td style="text-align: right;">0.304</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">16270</td>
<td style="text-align: right;">26031</td>
<td style="text-align: right;">25340</td>
<td style="text-align: right;">0.385</td>
<td style="text-align: right;">0.391</td>
<td style="text-align: right;">0.388</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">11793</td>
<td style="text-align: right;">10106</td>
<td style="text-align: right;">18177</td>
<td style="text-align: right;">0.539</td>
<td style="text-align: right;">0.393</td>
<td style="text-align: right;">0.455</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4483</td>
<td style="text-align: right;">4659</td>
<td style="text-align: right;">5385</td>
<td style="text-align: right;">0.490</td>
<td style="text-align: right;">0.454</td>
<td style="text-align: right;">0.472</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2928</td>
<td style="text-align: right;">930</td>
<td style="text-align: right;">1659</td>
<td style="text-align: right;">0.759</td>
<td style="text-align: right;">0.638</td>
<td style="text-align: right;">0.693</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:svpop-regions"></a>
<table>
<caption><span>Table S5:</span> Calling evaluation on the SVPOP dataset in different sets of regions for the HG5014 individual. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">TP</th>
<th style="text-align: right;">FP</th>
<th style="text-align: right;">FN</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">8618</td>
<td style="text-align: right;">7237</td>
<td style="text-align: right;">5416</td>
<td style="text-align: right;">0.546</td>
<td style="text-align: right;">0.614</td>
<td style="text-align: right;">0.578</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">4762</td>
<td style="text-align: right;">2048</td>
<td style="text-align: right;">5145</td>
<td style="text-align: right;">0.696</td>
<td style="text-align: right;">0.481</td>
<td style="text-align: right;">0.569</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">54</td>
<td style="text-align: right;">0.579</td>
<td style="text-align: right;">0.169</td>
<td style="text-align: right;">0.262</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">6176</td>
<td style="text-align: right;">6923</td>
<td style="text-align: right;">4678</td>
<td style="text-align: right;">0.475</td>
<td style="text-align: right;">0.569</td>
<td style="text-align: right;">0.518</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2428</td>
<td style="text-align: right;">1701</td>
<td style="text-align: right;">4542</td>
<td style="text-align: right;">0.584</td>
<td style="text-align: right;">0.348</td>
<td style="text-align: right;">0.436</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.143</td>
<td style="text-align: right;">0.222</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">2677</td>
<td style="text-align: right;">987</td>
<td style="text-align: right;">514</td>
<td style="text-align: right;">0.731</td>
<td style="text-align: right;">0.839</td>
<td style="text-align: right;">0.781</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">1180</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">321</td>
<td style="text-align: right;">0.869</td>
<td style="text-align: right;">0.786</td>
<td style="text-align: right;">0.825</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">0.636</td>
<td style="text-align: right;">0.259</td>
<td style="text-align: right;">0.368</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">3410</td>
<td style="text-align: right;">3789</td>
<td style="text-align: right;">2108</td>
<td style="text-align: right;">0.478</td>
<td style="text-align: right;">0.618</td>
<td style="text-align: right;">0.539</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2544</td>
<td style="text-align: right;">1092</td>
<td style="text-align: right;">1518</td>
<td style="text-align: right;">0.699</td>
<td style="text-align: right;">0.626</td>
<td style="text-align: right;">0.661</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.133</td>
<td style="text-align: right;">0.210</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">not called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4838</td>
<td style="text-align: right;">542</td>
<td style="text-align: right;">3678</td>
<td style="text-align: right;">0.899</td>
<td style="text-align: right;">0.568</td>
<td style="text-align: right;">0.696</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2034</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">3723</td>
<td style="text-align: right;">0.987</td>
<td style="text-align: right;">0.353</td>
<td style="text-align: right;">0.520</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">5245</td>
<td style="text-align: right;">8563</td>
<td style="text-align: right;">8789</td>
<td style="text-align: right;">0.394</td>
<td style="text-align: right;">0.374</td>
<td style="text-align: right;">0.384</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3741</td>
<td style="text-align: right;">3382</td>
<td style="text-align: right;">6166</td>
<td style="text-align: right;">0.533</td>
<td style="text-align: right;">0.378</td>
<td style="text-align: right;">0.442</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">3848</td>
<td style="text-align: right;">7125</td>
<td style="text-align: right;">7006</td>
<td style="text-align: right;">0.368</td>
<td style="text-align: right;">0.354</td>
<td style="text-align: right;">0.361</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">1990</td>
<td style="text-align: right;">2832</td>
<td style="text-align: right;">4980</td>
<td style="text-align: right;">0.426</td>
<td style="text-align: right;">0.286</td>
<td style="text-align: right;">0.342</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">1396</td>
<td style="text-align: right;">1468</td>
<td style="text-align: right;">1795</td>
<td style="text-align: right;">0.493</td>
<td style="text-align: right;">0.438</td>
<td style="text-align: right;">0.464</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">901</td>
<td style="text-align: right;">308</td>
<td style="text-align: right;">600</td>
<td style="text-align: right;">0.745</td>
<td style="text-align: right;">0.600</td>
<td style="text-align: right;">0.665</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4343</td>
<td style="text-align: right;">5595</td>
<td style="text-align: right;">1175</td>
<td style="text-align: right;">0.445</td>
<td style="text-align: right;">0.787</td>
<td style="text-align: right;">0.569</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3227</td>
<td style="text-align: right;">2451</td>
<td style="text-align: right;">835</td>
<td style="text-align: right;">0.573</td>
<td style="text-align: right;">0.794</td>
<td style="text-align: right;">0.666</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">not called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">116</td>
<td style="text-align: right;">109</td>
<td style="text-align: right;">8400</td>
<td style="text-align: right;">0.551</td>
<td style="text-align: right;">0.014</td>
<td style="text-align: right;">0.026</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">206</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">5551</td>
<td style="text-align: right;">0.911</td>
<td style="text-align: right;">0.036</td>
<td style="text-align: right;">0.069</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:simerror-bkpt"></a>
<table>
<caption><span>Table S6:</span> Breakpoint fine-tuning using graph augmentation from the read alignment. For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. In all cases, the errors affected 1-10 bp. </caption>
<colgroup>
<col style="width: 10%" />
<col style="width: 15%" />
<col style="width: 13%" />
<col style="width: 10%" />
<col style="width: 13%" />
<col style="width: 17%" />
<col style="width: 19%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">SV type</th>
<th style="text-align: left;">Error type</th>
<th style="text-align: left;">Breakpoint</th>
<th style="text-align: right;">Variant</th>
<th style="text-align: right;">Proportion</th>
<th style="text-align: right;">Mean size (bp)</th>
<th style="text-align: right;">Mean error (bp)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">one end</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">220</td>
<td style="text-align: right;">0.219</td>
<td style="text-align: right;">422.655</td>
<td style="text-align: right;">6.095</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">784</td>
<td style="text-align: right;">0.781</td>
<td style="text-align: right;">670.518</td>
<td style="text-align: right;">5.430</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">both ends</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">811</td>
<td style="text-align: right;">0.814</td>
<td style="text-align: right;">826.070</td>
<td style="text-align: right;">6.275</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">185</td>
<td style="text-align: right;">0.186</td>
<td style="text-align: right;">586.676</td>
<td style="text-align: right;">2.232</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INS</td>
<td style="text-align: left;">location/seq</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">123</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">428.724</td>
<td style="text-align: right;">6.667</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">1877</td>
<td style="text-align: right;">0.938</td>
<td style="text-align: right;">440.043</td>
<td style="text-align: right;">6.439</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INV</td>
<td style="text-align: left;">one end</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">868</td>
<td style="text-align: right;">0.835</td>
<td style="text-align: right;">762.673</td>
<td style="text-align: right;">5.161</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">172</td>
<td style="text-align: right;">0.165</td>
<td style="text-align: right;">130.244</td>
<td style="text-align: right;">5.884</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">both ends</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">950</td>
<td style="text-align: right;">0.992</td>
<td style="text-align: right;">556.274</td>
<td style="text-align: right;">5.624</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">0.008</td>
<td style="text-align: right;">200.000</td>
<td style="text-align: right;">1.375</td>
</tr>
</tbody>
</table>
<h3 id="supplementary-figures" class="page_break_before">Supplementary Figures</h3>
<figure>
<img src="images/hgsvc-sim-geno.png" alt="Figure S1: Genotyping evaluation on the HGSVC dataset using simulated reads. Reads were simulated from the HG00514 individual. The bottom panel zooms on the part highlighted by a dotted rectangle." id="fig:hgsvc-sim-geno" data-tag="S1" /><figcaption><span>Figure S1:</span> <strong>Genotyping evaluation on the HGSVC dataset using simulated reads.</strong> Reads were simulated from the HG00514 individual. The bottom panel zooms on the part highlighted by a dotted rectangle.</figcaption>
</figure>
<figure>
<img src="images/hgsvc-sim.png" alt="Figure S2: Calling evaluation on the HGSVC dataset using simulated reads. Reads were simulated from the HG00514 individual. The bottom panel zooms on the part highlighted by a dotted rectangle." id="fig:hgsvc-sim" data-tag="S2" /><figcaption><span>Figure S2:</span> <strong>Calling evaluation on the HGSVC dataset using simulated reads.</strong> Reads were simulated from the HG00514 individual. The bottom panel zooms on the part highlighted by a dotted rectangle.</figcaption>
</figure>
<figure>
<img src="images/hgsvc-real-geno.png" alt="Figure S3: Genotyping evaluation on the HGSVC dataset using real reads. Combined results across the HG00514, HG00733 and NA19240." id="fig:hgsvc-real-geno" data-tag="S3" /><figcaption><span>Figure S3:</span> <strong>Genotyping evaluation on the HGSVC dataset using real reads.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/hgsvc-real.png" alt="Figure S4: Calling evaluation on the HGSVC dataset using real reads. Combined results across the HG00514, HG00733 and NA19240." id="fig:hgsvc-real" data-tag="S4" /><figcaption><span>Figure S4:</span> <strong>Calling evaluation on the HGSVC dataset using real reads.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/giab5-geno.png" alt="Figure S5: Genotyping evaluation on the Genome in a Bottle dataset. Predicted genotypes on HG002 were compared to the high-quality SVs from this same individual." id="fig:giab-geno" data-tag="S5" /><figcaption><span>Figure S5:</span> <strong>Genotyping evaluation on the Genome in a Bottle dataset.</strong> Predicted genotypes on HG002 were compared to the high-quality SVs from this same individual.</figcaption>
</figure>
<figure>
<img src="images/giab5.png" alt="Figure S6: Calling evaluation on the Genome in a Bottle dataset. Calls on HG002 were compared to the high-quality SVs from this same individual." id="fig:giab" data-tag="S6" /><figcaption><span>Figure S6:</span> <strong>Calling evaluation on the Genome in a Bottle dataset.</strong> Calls on HG002 were compared to the high-quality SVs from this same individual.</figcaption>
</figure>
<figure>
<img src="images/chmpd-geno.png" alt="Figure S7: Genotyping evaluation on the CHM pseudo-diploid dataset. The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.[5] The bottom panel zooms on the part highlighted by a dotted rectangle." id="fig:chmpd-geno" data-tag="S7" /><figcaption><span>Figure S7:</span> <strong>Genotyping evaluation on the CHM pseudo-diploid dataset.</strong> The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span> The bottom panel zooms on the part highlighted by a dotted rectangle.</figcaption>
</figure>
<figure>
<img src="images/chmpd.png" alt="Figure S8: Calling evaluation on the CHM pseudo-diploid dataset. The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.[5]" id="fig:chmpd" data-tag="S8" /><figcaption><span>Figure S8:</span> <strong>Calling evaluation on the CHM pseudo-diploid dataset.</strong> The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span></figcaption>
</figure>
<figure>
<img src="images/svpop.png" alt="Figure S9: Calling evaluation on the SVPOP dataset. Combined results across the HG00514, HG00733 and NA19240." id="fig:svpop" data-tag="S9" /><figcaption><span>Figure S9:</span> <strong>Calling evaluation on the SVPOP dataset.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/svpop-regions.png" alt="Figure S10: Evaluation across different sets of regions in HG00514 (SVPOP dataset). Calling evaluation." id="fig:svpop-regions" data-tag="S10" /><figcaption><span>Figure S10:</span> <strong>Evaluation across different sets of regions in HG00514 (SVPOP dataset)</strong>. Calling evaluation.</figcaption>
</figure>
<figure>
<img src="images/hgsvc-giab-chmpd-svpop-best-f1-mincov90.png" alt="Figure S11: Benchmark summary when using a more stringent matching criterion. At least 90% coverage was necessary to consider a variant matched, instead of the 50% minimum coverage used in other figures." id="fig:eval-stringent" data-tag="S11" /><figcaption><span>Figure S11:</span> <strong>Benchmark summary when using a more stringent matching criterion</strong>. At least 90% coverage was necessary to consider a variant matched, instead of the 50% minimum coverage used in other figures.</figcaption>
</figure>
<figure>
<img src="images/simerror-bkpt-finetuning-vgcall.png" alt="Figure S12: Breakpoint fine-tuning using augmentation through “vg call”. For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. a) Proportion of variant for which breakpoints could be fine-tuned. b) Distribution of the amount of errors that could be corrected or not. c) Distribution of the size of the variants whose breakpoints could be fine-tuned or not." id="fig:simerror-bkpt" data-tag="S12" /><figcaption><span>Figure S12:</span> <strong>Breakpoint fine-tuning using augmentation through “vg call”.</strong> For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. a) Proportion of variant for which breakpoints could be fine-tuned. b) Distribution of the amount of errors that could be corrected or not. c) Distribution of the size of the variants whose breakpoints could be fine-tuned or not.</figcaption>
</figure>
<figure>
<img src="images/panel3.png" alt="Figure S13: Mapping comparison on graphs of the five strains set. Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the VCF graph (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs." id="fig:panel3" data-tag="S13" /><figcaption><span>Figure S13:</span> <strong>Mapping comparison on graphs of the <em>five strains set</em>.</strong> Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the VCF graph (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs.</figcaption>
</figure>
<figure>
<img src="images/panel5.png" alt="Figure S14: Mapping comparison on graphs of the all strains set. Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the VCF graph (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively." id="fig:panel5" data-tag="S14" /><figcaption><span>Figure S14:</span> <strong>Mapping comparison on graphs of the <em>all strains set</em>.</strong> Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the <em>cactus graph</em> (y-axis) and the <em>VCF graph</em> (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively.</figcaption>
</figure>
<figure>
<img src="images/panel6.png" alt="Figure S15: SV genotyping comparison using all reads. Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in the cactus graph and the VCF graph. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were aligned to the sample graphs and the quality of all alignments was used to ascertain SV genotyping performance. More accurate genotypes should result in sample graphs that have mappings with high identity and confidence for a greater proportion of the reads. a) Average delta in mapping identity of all short reads aligned to the sample graphs derived from cactus graph and VCF graph. b) Average delta in mapping quality of all short reads aligned to the sample graphs derived from cactus graph and VCF graph. Positive values denote an improvement of the cactus graph over the VCF graph. Colors represent the two strain sets and transparency indicates whether the respective strain was part of the five strains set." id="fig:panel6" data-tag="S15" /><figcaption><span>Figure S15:</span> <strong>SV genotyping comparison using all reads.</strong> Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in the <em>cactus graph</em> and the <em>VCF graph</em>. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were aligned to the sample graphs and the quality of all alignments was used to ascertain SV genotyping performance. More accurate genotypes should result in sample graphs that have mappings with high identity and confidence for a greater proportion of the reads. a) Average delta in mapping identity of all short reads aligned to the sample graphs derived from <em>cactus graph</em> and <em>VCF graph</em>. b) Average delta in mapping quality of all short reads aligned to the sample graphs derived from <em>cactus graph</em> and <em>VCF graph</em>. Positive values denote an improvement of the <em>cactus graph</em> over the <em>VCF graph</em>. Colors represent the two strain sets and transparency indicates whether the respective strain was part of the <em>five strains set</em>.</figcaption>
</figure>
<figure>
<img src="images/sveval-cartoon.png" alt="Figure S16: Overview of the SV evaluation by the sveval package. For deletions and inversions, we compute the proportion of a variant that is covered by variants in the other set, considering only variants overlapping with at least 10% reciprocal overlap. A variant is considered true positive if this coverage proportion is higher than 50% and false-positive or false-negative otherwise. A similar approach is used for insertions, although they are first clustered into pairs located less than 20 bp from each other. Then their inserted sequences are aligned to derive the coverage statistics. The SV evaluation approach is described in more detail in the Methods." id="fig:sveval" data-tag="S16" /><figcaption><span>Figure S16:</span> <strong>Overview of the SV evaluation by the <em>sveval</em> package</strong>. For deletions and inversions, we compute the proportion of a variant that is covered by variants in the other set, considering only variants overlapping with at least 10% reciprocal overlap. A variant is considered true positive if this coverage proportion is higher than 50% and false-positive or false-negative otherwise. A similar approach is used for insertions, although they are first clustered into pairs located less than 20 bp from each other. Then their inserted sequences are aligned to derive the coverage statistics. The SV evaluation approach is described in more detail in the <a href="#toil-vg-sveval">Methods</a>.</figcaption>
</figure>
<h3 id="supplementary-information" class="page_break_before">Supplementary Information</h3>
<h4 id="variation-graph-and-structural-variation">Variation graph and structural variation</h4>
<p>A variation graph encodes DNA sequence in its nodes. Such graphs are bidirected, in that we distinguish between edges incident on the starts of nodes from those incident on their ends. A path in such a graph is an ordered list of nodes where each is associated with an orientation. If a path walks from, for example, node A in the forward orientation to node B in the reverse orientation, then an edge must exist from the end of node A to the end of node B. Concatenating the sequences on each node in the path, taking the reverse complement when the node is visited in reverse orientation, produces a DNA sequence. Accordingly, variation graphs are constructed so as to encode haplotype sequences as walks through the graph. Variation between sequences shows up as bubbles in the graph <span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">24</a>]</span>.</p>
<h4 id="breakpoint-fine-tuning">Breakpoint fine-tuning</h4>
<p>In addition to genotyping, vg can use an augmentation step to modify the graph based on the read alignment and discover novel variants. On the simulated SVs from Figure <a href="#fig:1">1</a>b, this approach was able to correct many of the 1-10 bp breakpoint errors that were added to the input VCF. The breakpoints were accurately fine-tuned for 93.8% of the insertions (Figure <a href="#fig:simerror-bkpt">S12</a>a and Table <a href="#tbl:simerror-bkpt">S6</a>). For deletions, 78.1% of the variants were corrected when only one breakpoint had an error. In situations where both breakpoints of the deletions were incorrect, only 18.6% were corrected through graph augmentation, and only when the amount of error was small (Figure <a href="#fig:simerror-bkpt">S12</a>b). The breakpoints of less than 20% of the inversions could be corrected. Across all SV types, the size of the variant didn’t affect the ability to fine-tune the breakpoints through graph augmentation (Figure <a href="#fig:simerror-bkpt">S12</a>c).</p>
<h4 id="mappability-comparison-between-yeast-graphs">Mappability comparison between yeast graphs</h4>
<p>In order to elucidate whether the <em>cactus graph</em> represents the sequence diversity among the yeast strains better than the <em>VCF graph</em>, we mapped Illumina short reads to both graphs using <code>vg map</code>. Generally, more reads mapped to the <em>cactus graph</em> with high identity (Figures <a href="#fig:panel3">S13</a>a and <a href="#fig:panel5">S14</a>a) and high mapping quality (Figures <a href="#fig:panel3">S13</a>b and <a href="#fig:panel5">S14</a>b) than to the <em>VCF graph</em>. The <em>VCF graph</em> exhibited higher mappability only on the reference strain <em>S.c. S288C</em> with a marginal difference. The benefit of using the <em>cactus graph</em> is largest for strains in the <em>S. paradoxus</em> clade and smaller for strains in the <em>S. cerevisiae</em> clade. We found that the genetic distance to the reference strain (as estimated using Mash v2.1 <span class="citation" data-cites="mH9pzoIn">[<a href="#ref-mH9pzoIn">41</a>]</span>) correlated with the increase in confidently mapped reads (mapping quality &gt;= 60) between the <em>cactus graph</em> and the <em>VCF graph</em> (Spearman’s rank correlation, p-value=3.993e-06). These results suggest that the improvement in mappability is not driven by the higher sequence content in the <em>cactus graph</em> alone (16.8 / 15.4 Mb in the <em>cactus graph</em> compared to 12.6 / 12.4 Mb in the <em>VCF graph</em> for the <em>all strains set</em> and the <em>five strains set</em>, respectively). Instead, an explanation could be the construction of the <em>VCF graph</em> from a comprehensive but still limited list of variants and the lack of SNPs and small Indels in this list. Consequently, substantially fewer reads mapped to the <em>VCF graph</em> with perfect identity (Figures <a href="#fig:panel3">S13</a>a and <a href="#fig:panel5">S14</a>a, percent identity threshold = 100%) than to the <em>cactus graph</em>. The <em>cactus graph</em> has the advantage of implicitly incorporating variants of all types and sizes from the <em>de novo</em> assemblies. As a consequence, the <em>cactus graph</em> captures the genetic makeup of each strain more comprehensively and enables more reads to be mapped.</p>
<p>Interestingly, our measurements for the <em>five strains set</em> showed only small differences between the five strains that were used to construct the graph and the other seven strains (Figure <a href="#fig:panel3">S13</a>). Only the number of alignments with perfect identity is substantially lower for the strains that were not included in the creation of the graphs (Figure <a href="#fig:panel3">S13</a>a).</p>
<h4 id="running-time-comparison-between-different-tools-for-hg00514-as-genotyped-on-the-hgsvc-dataset">Running time comparison between different tools for HG00514 as genotyped on the HGSVC dataset</h4>
<a name="tbl:timing"></a>
<table>
<caption><span>Table S7:</span> Compute resources required for analysis of sample HG00514 on the HGSVC dataset. </caption>
<thead>
<tr class="header">
<th>Tool</th>
<th>Wall Time (m)</th>
<th>Cores</th>
<th>Nodes</th>
<th>Max Memory (G)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>vg</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>      vg construction</td>
<td>49</td>
<td>8</td>
<td>1 i3.8xlarge</td>
<td>0.4</td>
</tr>
<tr class="odd">
<td>      xg index</td>
<td>13</td>
<td>8</td>
<td>1 i3.8xlarge</td>
<td>48</td>
</tr>
<tr class="even">
<td>      snarls index</td>
<td>23</td>
<td>1</td>
<td>50 i3.8xlarge</td>
<td>17</td>
</tr>
<tr class="odd">
<td>      gcsa2 index</td>
<td>792</td>
<td>16</td>
<td>1 i3.8xlarge</td>
<td>45</td>
</tr>
<tr class="even">
<td>      mapping</td>
<td>177</td>
<td>32</td>
<td>50 r3.8xlarge</td>
<td>32</td>
</tr>
<tr class="odd">
<td>      genotyping (pack + call)</td>
<td>56</td>
<td>10</td>
<td>1 i3.4xlarge</td>
<td>63</td>
</tr>
<tr class="even">
<td><strong>BayesTyper</strong></td>
<td>94</td>
<td>24</td>
<td>1 i3.8xlarge</td>
<td>119</td>
</tr>
<tr class="odd">
<td><strong>bwa mem</strong></td>
<td>240</td>
<td>32</td>
<td>1 i3.8xlarge</td>
<td>14</td>
</tr>
<tr class="even">
<td>      <strong>Delly</strong></td>
<td>69</td>
<td>1</td>
<td>1 i3.8xlarge</td>
<td>69</td>
</tr>
<tr class="odd">
<td>      <strong>SVTyper</strong></td>
<td>477</td>
<td>1</td>
<td>1 i3.8xlarge</td>
<td>0.7</td>
</tr>
<tr class="even">
<td>      <strong>Paragraph</strong></td>
<td>76</td>
<td>32</td>
<td>1 i3.8xlarge</td>
<td>5.9</td>
</tr>
</tbody>
</table>
<p>SMRT-SV2 required roughly 36 hours and 30G ram on 30 cores to genotype the three HGSVC samples on the “SVPOP” VCF. These numbers are not directly comparable to the above table because 1) they apply to the “SVPOP” rather than “HGSVC” dataset (upon which we were unable to run SMRT-SV2) and 2) we were unable to install SMRT-SV2 on AWS nodes and ran it on an older, shared server at UCSC instead.</p>
<p>Delly, SVTyper and Paragraph start from a set of aligned reads, hence we also show the running time for read alignment with <code>bwa mem</code>. For BayesTyper, the numbers include both khmer counting with KMC and genotyping.</p>
<p>Note: toil-vg reserves 200G memory by default for <code>vg snarls</code>. For this graph, about an order of magnitude less was required. It could have been run on 10 cores on 5 nodes instead.</p>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-2gpKwL67">
<p>1. <strong>The impact of structural variation on human gene expression</strong><br />
Colby ChiangAlexandra J Scott, Joe R Davis, Emily K Tsang, Xin Li, Yungil Kim, Tarik Hadzic, Farhan N Damani, Liron Ganel, … Ira M Hall<br />
<em>Nature Genetics</em> (2017-04-03) <a href="https://doi.org/f9xvr6">https://doi.org/f9xvr6</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3834">10.1038/ng.3834</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28369037">28369037</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5406250">PMC5406250</a></p>
</div>
<div id="ref-ebc66eBr">
<p>2. <strong>Phenotypic impact of genomic structural variation: insights from and for human disease</strong><br />
Joachim Weischenfeldt, Orsolya Symmons, François Spitz, Jan O. Korbel<br />
<em>Nature Reviews Genetics</em> (2013-01-18) <a href="https://doi.org/f4nhxh">https://doi.org/f4nhxh</a><br />
DOI: <a href="https://doi.org/10.1038/nrg3373">10.1038/nrg3373</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23329113">23329113</a></p>
</div>
<div id="ref-AltPnocw">
<p>3. <strong>SpeedSeq: ultra-fast personal genome analysis and interpretation</strong><br />
Colby Chiang, Ryan M Layer, Gregory G Faust, Michael R Lindberg, David B Rose, Erik P Garrison, Gabor T Marth, Aaron R Quinlan, Ira M Hall<br />
<em>Nature Methods</em> (2015-08-10) <a href="https://doi.org/gcpgfh">https://doi.org/gcpgfh</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.3505">10.1038/nmeth.3505</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26258291">26258291</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4589466">PMC4589466</a></p>
</div>
<div id="ref-nLvQCjXU">
<p>4. <strong>DELLY: structural variant discovery by integrated paired-end and split-read analysis</strong><br />
T. Rausch, T. Zichner, A. Schlattl, A. M. Stutz, V. Benes, J. O. Korbel<br />
<em>Bioinformatics</em> (2012-09-07) <a href="https://doi.org/f38r2c">https://doi.org/f38r2c</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bts378">10.1093/bioinformatics/bts378</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22962449">22962449</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3436805">PMC3436805</a></p>
</div>
<div id="ref-3NNFS6U2">
<p>5. <strong>Characterizing the Major Structural Variant Alleles of the Human Genome</strong><br />
Peter A. Audano, Arvis Sulovari, Tina A. Graves-Lindsay, Stuart Cantsilieris, Melanie Sorensen, AnneMarie E. Welch, Max L. Dougherty, Bradley J. Nelson, Ankeeta Shah, Susan K. Dutcher, … Evan E. Eichler<br />
<em>Cell</em> (2019-01) <a href="https://doi.org/gfthvz">https://doi.org/gfthvz</a><br />
DOI: <a href="https://doi.org/10.1016/j.cell.2018.12.019">10.1016/j.cell.2018.12.019</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661756">30661756</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6438697">PMC6438697</a></p>
</div>
<div id="ref-qA6dWFP">
<p>6. <strong>An integrated map of structural variation in 2,504 human genomes</strong><br />
Peter H. SudmantTobias Rausch, Eugene J. Gardner, Robert E. Handsaker, Alexej Abyzov, John Huddleston, Yan Zhang, Kai Ye, Goo Jun, … Jan O. Korbel<br />
<em>Nature</em> (2015-09-30) <a href="https://doi.org/73c">https://doi.org/73c</a><br />
DOI: <a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26432246">26432246</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4617611">PMC4617611</a></p>
</div>
<div id="ref-py6BC5kj">
<p>7. <strong>Whole-genome sequence variation, population structure and demographic history of the Dutch population</strong><em>Nature Genetics</em> (2014-06-29) <a href="https://doi.org/f6bxm8">https://doi.org/f6bxm8</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3021">10.1038/ng.3021</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24974849">24974849</a></p>
</div>
<div id="ref-z91V6jjU">
<p>8. <strong>Resolving the complexity of the human genome using single-molecule sequencing</strong><br />
Mark J. P. Chaisson, John Huddleston, Megan Y. Dennis, Peter H. Sudmant, Maika Malig, Fereydoun Hormozdiari, Francesca Antonacci, Urvashi Surti, Richard Sandstrom, Matthew Boitano, … Evan E. Eichler<br />
<em>Nature</em> (2014-11-10) <a href="https://doi.org/w69">https://doi.org/w69</a><br />
DOI: <a href="https://doi.org/10.1038/nature13907">10.1038/nature13907</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25383537">25383537</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4317254">PMC4317254</a></p>
</div>
<div id="ref-rs7e40wC">
<p>9. <strong>Discovery and genotyping of structural variation from long-read haploid genome sequence data</strong><br />
John Huddleston, Mark J.P. Chaisson, Karyn Meltz Steinberg, Wes Warren, Kendra Hoekzema, David Gordon, Tina A. Graves-Lindsay, Katherine M. Munson, Zev N. Kronenberg, Laura Vives, … Evan E. Eichler<br />
<em>Genome Research</em> (2016-11-28) <a href="https://doi.org/f9x79h">https://doi.org/f9x79h</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214007.116">10.1101/gr.214007.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27895111">27895111</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411763">PMC5411763</a></p>
</div>
<div id="ref-PRx3qEIm">
<p>10. <strong>Mapping and phasing of structural variation in patient genomes using nanopore sequencing</strong><br />
Mircea Cretu Stancu, Markus J. van Roosmalen, Ivo Renkens, Marleen M. Nieboer, Sjors Middelkamp, Joep de Ligt, Giulia Pregno, Daniela Giachino, Giorgia Mandrile, Jose Espejo Valle-Inclan, … Wigard P. Kloosterman<br />
<em>Nature Communications</em> (2017-11-06) <a href="https://doi.org/gftpt9">https://doi.org/gftpt9</a><br />
DOI: <a href="https://doi.org/10.1038/s41467-017-01343-4">10.1038/s41467-017-01343-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29109544">29109544</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5673902">PMC5673902</a></p>
</div>
<div id="ref-121OWcTA4">
<p>11. <strong>Genome-wide reconstruction of complex structural variants using read clouds</strong><br />
Noah Spies, Ziming Weng, Alex Bishara, Jennifer McDaniel, David Catoe, Justin M Zook, Marc Salit, Robert B West, Serafim Batzoglou, Arend Sidow<br />
<em>Nature Methods</em> (2017-07-17) <a href="https://doi.org/gbnhwk">https://doi.org/gbnhwk</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4366">10.1038/nmeth.4366</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28714986">28714986</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5578891">PMC5578891</a></p>
</div>
<div id="ref-6KbgcueR">
<p>12. <strong>Nanopore sequencing and assembly of a human genome with ultra-long reads</strong><br />
Miten Jain, Sergey Koren, Karen H Miga, Josh Quick, Arthur C Rand, Thomas A Sasani, John R Tyson, Andrew D Beggs, Alexander T Dilthey, Ian T Fiddes, … Matthew Loose<br />
<em>Nature Biotechnology</em> (2018-01-29) <a href="https://doi.org/gczffw">https://doi.org/gczffw</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4060">10.1038/nbt.4060</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29431738">29431738</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5889714">PMC5889714</a></p>
</div>
<div id="ref-Pu6SY37C">
<p>13. <strong>Phased diploid genome assembly with single-molecule real-time sequencing</strong><br />
Chen-Shan Chin, Paul Peluso, Fritz J Sedlazeck, Maria Nattestad, Gregory T Concepcion, Alicia Clum, Christopher Dunn, Ronan O’Malley, Rosa Figueroa-Balderas, Abraham Morales-Cruz, … Michael C Schatz<br />
<em>Nature Methods</em> (2016-10-17) <a href="https://doi.org/f9fv4w">https://doi.org/f9fv4w</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4035">10.1038/nmeth.4035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27749838">27749838</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5503144">PMC5503144</a></p>
</div>
<div id="ref-Qa8mx6Ll">
<p>14. <strong>Genome graphs and the evolution of genome inference</strong><br />
Benedict Paten, Adam M. Novak, Jordan M. Eizenga, Erik Garrison<br />
<em>Genome Research</em> (2017-03-30) <a href="https://doi.org/f95nhd">https://doi.org/f95nhd</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214155.116">10.1101/gr.214155.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28360232">28360232</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411762">PMC5411762</a></p>
</div>
<div id="ref-10jxt15v0">
<p>15. <strong>Variation graph toolkit improves read mapping by representing genetic variation in the reference</strong><br />
Erik Garrison, Jouni Sirén, Adam M Novak, Glenn Hickey, Jordan M Eizenga, Eric T Dawson, William Jones, Shilpa Garg, Charles Markello, Michael F Lin, … Richard Durbin<br />
<em>Nature Biotechnology</em> (2018-08-20) <a href="https://doi.org/gd2zqs">https://doi.org/gd2zqs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4227">10.1038/nbt.4227</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30125266">30125266</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6126949">PMC6126949</a></p>
</div>
<div id="ref-DuODeStx">
<p>16. <strong>Genome Graphs</strong><br />
Adam M. Novak, Glenn Hickey, Erik Garrison, Sean Blum, Abram Connelly, Alexander Dilthey, Jordan Eizenga, M. A. Saleh Elmohamed, Sally Guthrie, André Kahles, … Benedict Paten<br />
<em>Cold Spring Harbor Laboratory</em> (2017-01-18) <a href="https://doi.org/gdcc74">https://doi.org/gdcc74</a><br />
DOI: <a href="https://doi.org/10.1101/101378">10.1101/101378</a></p>
</div>
<div id="ref-11Jy8B61m">
<p>17. <strong>Fast and accurate genomic analyses using genome graphs</strong><br />
Goran Rakocevic, Vladimir Semenyuk, Wan-Ping Lee, James Spencer, John Browning, Ivan J. Johnson, Vladan Arsenijevic, Jelena Nadj, Kaushik Ghose, Maria C. Suciu, … Deniz Kural<br />
<em>Nature Genetics</em> (2019-01-14) <a href="https://doi.org/gftd46">https://doi.org/gftd46</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0316-4">10.1038/s41588-018-0316-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30643257">30643257</a></p>
</div>
<div id="ref-ohTIiqfV">
<p>18. <strong>Graphtyper enables population-scale genotyping using pangenome graphs</strong><br />
Hannes P Eggertsson, Hakon Jonsson, Snaedis Kristmundsdottir, Eirikur Hjartarson, Birte Kehr, Gisli Masson, Florian Zink, Kristjan E Hjorleifsson, Aslaug Jonasdottir, Adalbjorg Jonasdottir, … Bjarni V Halldorsson<br />
<em>Nature Genetics</em> (2017-09-25) <a href="https://doi.org/gbx7v6">https://doi.org/gbx7v6</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3964">10.1038/ng.3964</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28945251">28945251</a></p>
</div>
<div id="ref-14Uxmwbxm">
<p>19. <strong>Accurate genotyping across variant classes and lengths using variant graphs</strong><br />
Jonas Andreas SibbesenLasse Maretty, Anders Krogh<br />
<em>Nature Genetics</em> (2018-06-18) <a href="https://doi.org/gdndnz">https://doi.org/gdndnz</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0145-5">10.1038/s41588-018-0145-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29915429">29915429</a></p>
</div>
<div id="ref-zjxFCnET">
<p>20. <strong>Paragraph: A graph-based structural variant genotyper for short-read sequence data</strong><br />
Sai Chen, Peter Krusche, Egor Dolzhenko, Rachel M. Sherman, Roman Petrovski, Felix Schlesinger, Melanie Kirsche, David R. Bentley, Michael C. Schatz, Fritz J. Sedlazeck, Michael A. Eberle<br />
<em>Cold Spring Harbor Laboratory</em> (2019-05-10) <a href="https://doi.org/gf9sdp">https://doi.org/gf9sdp</a><br />
DOI: <a href="https://doi.org/10.1101/635011">10.1101/635011</a></p>
</div>
<div id="ref-vQTymKCj">
<p>21. <strong>Multi-platform discovery of haplotype-resolved structural variation in human genomes</strong><br />
Mark J.P. Chaisson, Ashley D. Sanders, Xuefang Zhao, Ankit Malhotra, David Porubsky, Tobias Rausch, Eugene J. Gardner, Oscar Rodriguez, Li Guo, Ryan L. Collins, … Charles Lee<br />
<em>Cold Spring Harbor Laboratory</em> (2017-09-23) <a href="https://doi.org/gftxhc">https://doi.org/gftxhc</a><br />
DOI: <a href="https://doi.org/10.1101/193144">10.1101/193144</a></p>
</div>
<div id="ref-RCzDkVqI">
<p>22. <strong>A robust benchmark for germline structural variant detection</strong><br />
Justin M. Zook, Nancy F. Hansen, Nathan D. Olson, Lesley M. Chapman, James C. Mullikin, Chunlin Xiao, Stephen Sherry, Sergey Koren, Adam M. Phillippy, Paul C. Boutros, … <br />
<em>Cold Spring Harbor Laboratory</em> (2019-06-09) <a href="https://doi.org/gf3tpt">https://doi.org/gf3tpt</a><br />
DOI: <a href="https://doi.org/10.1101/664623">10.1101/664623</a></p>
</div>
<div id="ref-16GvGhO20">
<p>23. <strong>Reproducible integration of multiple sequencing datasets to form high-confidence SNP, indel, and reference calls for five human genome reference materials</strong><br />
Justin M. Zook, Jennifer McDaniel, Hemang Parikh, Haynes Heaton, Sean A. Irvine, Len Trigg, Rebecca Truty, Cory Y. McLean, Francisco M. De La Vega, Chunlin Xiao, … Marc Salit<br />
<em>Cold Spring Harbor Laboratory</em> (2018-03-13) <a href="https://doi.org/gfwsmj">https://doi.org/gfwsmj</a><br />
DOI: <a href="https://doi.org/10.1101/281006">10.1101/281006</a></p>
</div>
<div id="ref-xJlNnKH2">
<p>24. <strong>Superbubbles, Ultrabubbles, and Cacti</strong><br />
Benedict Paten, Jordan M. Eizenga, Yohei M. Rosen, Adam M. Novak, Erik Garrison, Glenn Hickey<br />
<em>Journal of Computational Biology</em> (2018-07) <a href="https://doi.org/gdw582">https://doi.org/gdw582</a><br />
DOI: <a href="https://doi.org/10.1089/cmb.2017.0251">10.1089/cmb.2017.0251</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29461862">29461862</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6067107">PMC6067107</a></p>
</div>
<div id="ref-14neTdqfN">
<p>25. <strong>Extensive sequencing of seven human genomes to characterize benchmark reference materials</strong><br />
Justin M. Zook, David Catoe, Jennifer McDaniel, Lindsay Vang, Noah Spies, Arend Sidow, Ziming Weng, Yuling Liu, Christopher E. Mason, Noah Alexander, … Marc Salit<br />
<em>Scientific Data</em> (2016-06-07) <a href="https://doi.org/f84nqc">https://doi.org/f84nqc</a><br />
DOI: <a href="https://doi.org/10.1038/sdata.2016.25">10.1038/sdata.2016.25</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27271295">27271295</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4896128">PMC4896128</a></p>
</div>
<div id="ref-1FgS53pXi">
<p>26. <strong>Cactus: Algorithms for genome multiple sequence alignment</strong><br />
B. Paten, D. Earl, N. Nguyen, M. Diekhans, D. Zerbino, D. Haussler<br />
<em>Genome Research</em> (2011-06-10) <a href="https://doi.org/bk4697">https://doi.org/bk4697</a><br />
DOI: <a href="https://doi.org/10.1101/gr.123356.111">10.1101/gr.123356.111</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21665927">21665927</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3166836">PMC3166836</a></p>
</div>
<div id="ref-7f5OKa5O">
<p>27. <strong>Contrasting evolutionary genome dynamics between domesticated and wild yeasts</strong><br />
Jia-Xing Yue, Jing Li, Louise Aigrain, Johan Hallin, Karl Persson, Karen Oliver, Anders Bergström, Paul Coupland, Jonas Warringer, Marco Cosentino Lagomarsino, … Gianni Liti<br />
<em>Nature Genetics</em> (2017-04-17) <a href="https://doi.org/f93kpp">https://doi.org/f93kpp</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3847">10.1038/ng.3847</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28416820">28416820</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5446901">PMC5446901</a></p>
</div>
<div id="ref-krO7WgVi">
<p>28. <strong>Assemblytics: a web analytics tool for the detection of variants from an assembly</strong><br />
Maria Nattestad, Michael C. Schatz<br />
<em>Bioinformatics</em> (2016-06-17) <a href="https://doi.org/f9c485">https://doi.org/f9c485</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btw369">10.1093/bioinformatics/btw369</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27318204">27318204</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6191160">PMC6191160</a></p>
</div>
<div id="ref-oVaXIwl5">
<p>29. <strong>Discovery, genotyping and characterization of structural variation and novel sequence at single nucleotide resolution from de novo genome assemblies on a population scale</strong><br />
Siyang LiuShujia Huang, Junhua Rao, Weijian Ye, Anders Krogh, Jun Wang<br />
<em>GigaScience</em> (2015-12) <a href="https://doi.org/f75r4n">https://doi.org/f75r4n</a><br />
DOI: <a href="https://doi.org/10.1186/s13742-015-0103-4">10.1186/s13742-015-0103-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26705468">26705468</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4690232">PMC4690232</a></p>
</div>
<div id="ref-172cJaw4Q">
<p>30. <strong>Minimap2: pairwise alignment for nucleotide sequences</strong><br />
Heng Li<br />
<em>Bioinformatics</em> (2018-05-10) <a href="https://doi.org/gdhbqt">https://doi.org/gdhbqt</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29750242">29750242</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6137996">PMC6137996</a></p>
</div>
<div id="ref-10Jid8Wql">
<p>31. <strong>The Pancancer Analysis of Whole Genomes (PCAWG).</strong><a href="https://dcc.icgc.org/pcawg">https://dcc.icgc.org/pcawg</a></p>
</div>
<div id="ref-mWj2p7Xp">
<p>32. <strong>Genomics England 100,000 Genomes Project.</strong><a href="https://www.genomicsengland.co.uk">https://www.genomicsengland.co.uk</a></p>
</div>
<div id="ref-ir1O1h8n">
<p>33. <strong>Whole Genome Sequencing in the NHLBI Trans-Omics for Precision Medicine (TOPMed).</strong><a href="https://www.nhlbiwgs.org/">https://www.nhlbiwgs.org/</a></p>
</div>
<div id="ref-pJAv1D8R">
<p>34. <strong>Sequencing and de novo assembly of 150 genomes from Denmark as a population reference</strong><br />
Lasse Maretty, Jacob Malte Jensen, Bent Petersen, Jonas Andreas Sibbesen, Siyang Liu, Palle Villesen, Laurits Skov, Kirstine Belling, Christian Theil Have, Jose M. G. Izarzugaza, … Mikkel Heide Schierup<br />
<em>Nature</em> (2017-07-26) <a href="https://doi.org/gbpnnx">https://doi.org/gbpnnx</a><br />
DOI: <a href="https://doi.org/10.1038/nature23264">10.1038/nature23264</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28746312">28746312</a></p>
</div>
<div id="ref-faeC2cx0">
<p>35. <strong>Toil enables reproducible, open source, big biomedical data analyses</strong><br />
John Vivian, Arjun Arkal Rao, Frank Austin Nothaft, Christopher Ketchum, Joel Armstrong, Adam Novak, Jacob Pfeil, Jake Narkizian, Alden D Deran, Audrey Musselman-Brown, … Benedict Paten<br />
<em>Nature Biotechnology</em> (2017-04) <a href="https://doi.org/gfxbhs">https://doi.org/gfxbhs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.3772">10.1038/nbt.3772</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28398314">28398314</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5546205">PMC5546205</a></p>
</div>
<div id="ref-LAG2q9WK">
<p>36. <strong>Bcftools 1.9</strong><a href="https://samtools.github.io/bcftools/">https://samtools.github.io/bcftools/</a></p>
</div>
<div id="ref-NCr4QkOg">
<p>37. <strong>A framework for variation discovery and genotyping using next-generation DNA sequencing data</strong><br />
Mark A DePristo, Eric Banks, Ryan Poplin, Kiran V Garimella, Jared R Maguire, Christopher Hartl, Anthony A Philippakis, Guillermo del Angel, Manuel A Rivas, Matt Hanna, … Mark J Daly<br />
<em>Nature Genetics</em> (2011-04-10) <a href="https://doi.org/d9k453">https://doi.org/d9k453</a><br />
DOI: <a href="https://doi.org/10.1038/ng.806">10.1038/ng.806</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21478889">21478889</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3083463">PMC3083463</a></p>
</div>
<div id="ref-1DYmkalz4">
<p>38. <strong>Integrating mapping-, assembly- and haplotype-based approaches for calling variants in clinical sequencing applications</strong><br />
Andy RimmerHang Phan, Iain Mathieson, Zamin Iqbal, Stephen RF Twigg, Andrew OM Wilkie, Gil McVean, Gerton Lunter<br />
<em>Nature Genetics</em> (2014-07-13) <a href="https://doi.org/f6b6dk">https://doi.org/f6b6dk</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3036">10.1038/ng.3036</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25017105">25017105</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4753679">PMC4753679</a></p>
</div>
<div id="ref-jj4JJlsg">
<p>39. <strong>Haplotype-aware graph indexes</strong><br />
Jouni Sirén, Erik Garrison, Adam M. Novak, Benedict Paten, Richard Durbin<br />
<em>Schloss Dagstuhl - Leibniz-Zentrum fuer Informatik GmbH, Wadern/Saarbruecken, Germany</em> (2018) <a href="https://doi.org/gf2jss">https://doi.org/gf2jss</a><br />
DOI: <a href="https://doi.org/10.4230/lipics.wabi.2018.4">10.4230/lipics.wabi.2018.4</a></p>
</div>
<div id="ref-1HWiAHnIw">
<p>40. <strong>BEDTools: a flexible suite of utilities for comparing genomic features</strong><br />
Aaron R. Quinlan, Ira M. Hall<br />
<em>Bioinformatics</em> (2010-01-28) <a href="https://doi.org/cmrms3">https://doi.org/cmrms3</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btq033">10.1093/bioinformatics/btq033</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278">20110278</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2832824">PMC2832824</a></p>
</div>
<div id="ref-mH9pzoIn">
<p>41. <strong>Mash: fast genome and metagenome distance estimation using MinHash</strong><br />
Brian D. Ondov, Todd J. Treangen, Páll Melsted, Adam B. Mallonee, Nicholas H. Bergman, Sergey Koren, Adam M. Phillippy<br />
<em>Genome Biology</em> (2016-06-20) <a href="https://doi.org/gfx74q">https://doi.org/gfx74q</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-016-0997-x">10.1186/s13059-016-0997-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27323842">27323842</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4915045">PMC4915045</a></p>
</div>
<div id="ref-tIvRXd6o">
<p>42. <strong>PHYLIP - Phylogeny Inference Package (Version 3.2).</strong><br />
Joel Felsenstein<br />
<em>Cladistics</em> (1989)</p>
</div>
<div id="ref-rXSPQWTb">
<p>43. <strong>vgteam/sv-genotyping-paper: Code repository for “Genotyping structural variants in pangenome graphs using the vg toolkit”</strong><br />
Jean Monlong, Glenn Hickey, David Heller<br />
<em>Zenodo</em> (2019-05-29) <a href="https://doi.org/gf29bv">https://doi.org/gf29bv</a><br />
DOI: <a href="https://doi.org/10.5281/zenodo.3234891">10.5281/zenodo.3234891</a></p>
</div>
</div>
<!-- default theme -->

<style>
    /* import google fonts */
    @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Open Sans", "Helvetica", sans-serif;

        /* prevent text inflation on some mobile browsers */
        -webkit-text-size-adjust: none !important;
        -moz-text-size-adjust: none !important;
        -o-text-size-adjust: none !important;
        text-size-adjust: none !important;
    }

    @media only screen {
        /* "page" element */
        body {
            position: relative;
            box-sizing: border-box;
            font-size: 12pt;
            line-height: 1.5;
            max-width: 8.5in;
            margin: 20px auto;
            padding: 40px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            padding: 20px;
            margin: 0;
            border-radius: 0;
            border: none;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
            background: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin: 40px 0;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        border-bottom: solid 1px #bdbdbd;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 10px;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
    }

    /* normal links (not empty, not button link, not syntax highlighting link) */
    a:not(:empty):not(.button):not(.sourceLine) {
        padding-left: 1px;
        padding-right: 1px;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        padding-left: 20px;
    }

    /* class for styling text semibold */
    .semibold {
        font-weight: 600;
    }

    /* class for styling elements horizontally left aligned */
    .left {
        display: block;
        text-align: left;
        margin-left: auto;
        margin-right: 0;
        justify-content: left;
    }

    /* class for styling elements horizontally centered */
    .center {
        display: block;
        text-align: center;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }

    /* class for styling elements horizontally right aligned */
    .right {
        display: block;
        text-align: right;
        margin-left: 0;
        margin-right: auto;
        justify-content: right;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bdbdbd;
    }

    /* paragraphs, horizontal dividers, figures, tables, code */
    p,
    hr,
    figure,
    table,
    pre {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure */
    figure {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        padding: 0;
        padding-top: 10px;
    }

    /* figure image element */
    figure img {
        max-width: 100%;
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure auto-number */
    img + figcaption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bdbdbd;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        word-break: break-all;
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* table caption */
    caption {
        text-align: left;
        padding: 0;
        padding-bottom: 10px;
    }

    /* table auto-number */
    table > caption > span:first-of-type,
    div.table_wrapper > table > caption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        padding: 10px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
        break-inside: avoid;
        text-align: left;
    }

    /* inline code, ie code within normal text */
    :not(pre) > code {
        padding: 0 4px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bdbdbd;
        padding-left: 16px;
        break-inside: avoid;
    }

    /* -------------------------------------------------- */
    /* banners */
    /* -------------------------------------------------- */

    /* info banners */
    .banner {
        box-sizing: border-box;
        display: block;
        position: relative;
        width: 100%;
        margin-top: 20px;
        margin-bottom: 20px;
        padding: 20px;
        text-align: center;
    }

    /* paragraph in banner */
    .banner > p {
        margin: 0;
    }

    /* -------------------------------------------------- */
    /* highlight colors */
    /* -------------------------------------------------- */

    .white {
        background: #ffffff;
    }
    .lightgrey {
        background: #eeeeee;
    }
    .grey {
        background: #757575;
    }
    .darkgrey {
        background: #424242;
    }
    .black {
        background: #000000;
    }
    .lightred {
        background: #ffcdd2;
    }
    .lightyellow {
        background: #ffecb3;
    }
    .lightgreen {
        background: #dcedc8;
    }
    .lightblue {
        background: #e3f2fd;
    }
    .lightpurple {
        background: #f3e5f5;
    }
    .red {
        background: #f44336;
    }
    .orange {
        background: #ff9800;
    }
    .yellow {
        background: #ffeb3b;
    }
    .green {
        background: #4caf50;
    }
    .blue {
        background: #2196f3;
    }
    .purple {
        background: #9c27b0;
    }
    .white,
    .lightgrey,
    .lightred,
    .lightyellow,
    .lightgreen,
    .lightblue,
    .lightpurple,
    .orange,
    .yellow,
    .white a,
    .lightgrey a,
    .lightred a,
    .lightyellow a,
    .lightgreen a,
    .lightblue a,
    .lightpurple a,
    .orange a,
    .yellow a {
        color: #000000;
    }
    .grey,
    .darkgrey,
    .black,
    .red,
    .green,
    .blue,
    .purple,
    .grey a,
    .darkgrey a,
    .black a,
    .red a,
    .green a,
    .blue a,
    .purple a {
        color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 0.75em;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bdbdbd;
        border-radius: 5px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f5f5f5;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 5px;
        border: none;
        width: 20px;
        height: 20px;
        min-width: 20px;
        min-height: 20px;
    }

    /* icon button inner svg image */
    .icon_button > svg {
        height: 16px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* print control */
    /* -------------------------------------------------- */

    @media print {
        @page {
            /* suggested printing margin */
            margin: 0.5in;
        }

        /* document and "page" elements */
        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        /* "page" element */
        body {
            font-size: 11pt !important;
            line-height: 1.35;
        }

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 15px 0;
        }

        /* heading 1 */
        h1 {
            font-size: 1.75em;
        }

        /* heading 2 */
        h2 {
            font-size: 1.25em;
            margin-top: 0;
        }

        /* heading 3 */
        h3 {
            font-size: 1.10em;
        }

        /* figures and tables */
        figure, table {
            font-size: 0.85em;
        }

        /* table cells */
        th,
        td {
            padding: 5px;
        }

        /* shrink font awesome icons */
        i.fas,
        i.fab,
        i.far,
        i.fal {
            transform: scale(0.85);
        }

        /* decrease banner margins */
        .banner {
            margin-top: 15px;
            margin-bottom: 15px;
            padding: 15px;
        }

        /* class for centering an element vertically on its own page */
        .page_center {
            margin: auto;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            vertical-align: middle;
            break-before: page;
            break-after: page;
        }

        /* always insert a page break before the element */
        .page_break_before {
            break-before: page;
        }

        /* always insert a page break after the element */
        .page_break_after {
            break-after: page;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            break-before: avoid;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            break-after: avoid;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            break-inside: avoid;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

    div.sourceCode {
        /* prevent background fill on top-most code block  container */
        background: none !important;
    }

    .sourceCode * {
        /* force consistent line spacing */
        line-height: 1.5 !important;
    }

    div.sourceCode {
        /* style code block margins same as <pre> element */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* mathjax containers */
    .math.display > span:not(.MathJax_Preview) {
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        overflow-x: auto !important;
        overflow-y: hidden !important;
        justify-content: center;
        align-items: center;
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border-radius: 5px !important;
        border: solid 1px #bdbdbd !important;
        box-shadow: none !important;
    }

    /* equation auto-number */
    span[id^="eq:"] > span.math.display + span {
        font-weight: 600;
    }

    /* equation */
    span[id^="eq:"] > span.math.display > span {
        /* nudge to make room for equation auto-number */
        margin-right: 40px !important;
    }

    /* -------------------------------------------------- */
    /* table scroll plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* table wrapper */
        .table_wrapper {
            /* show scrollbar on tables if necessary to prevent overflow */
            overflow: auto;
            width: 100%;
            margin: 20px 0;
        }

        /* table within table wrapper */
        .table_wrapper table,
        .table_wrapper table * {
            /* don't break table words */
            word-break: normal !important;
        }

        .table_wrapper > table {
            /* move margins from table to table_wrapper to allow margin collapsing */
            margin: 0;
        }
    }

    /* -------------------------------------------------- */
    /* anchors plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anchor button */
        .anchor {
            opacity: 0;
            margin-left: 5px;
        }

        /* anchor buttons within <h2>'s */
        h2 .anchor {
            margin-left: 10px;
        }

        /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
        *:hover > .anchor,
        .anchor:hover,
        .anchor:focus {
            opacity: 1;
        }

        /* anchor button when hovered */
        .anchor:hover {
            cursor: pointer;
        }
    }

    /* always show anchor button on devices with no mouse/hover ability */
    @media (hover: none) {
        .anchor {
            opacity: 1;
        }
    }

    /* always hide anchor button on print */
    @media only print {
        .anchor {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* accordion plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* accordion arrow button */
        .accordion_arrow {
            margin-right: 10px;
        }

        /* arrow icon when <h2> data-collapsed attribute true */
        h2[data-collapsed="true"] > .accordion_arrow > svg {
            transform: rotate(-90deg);
        }

        /* all elements (except <h2>'s) when data-collapsed attribute true */
        *:not(h2)[data-collapsed="true"] {
            display: none;
        }

        /* accordion arrow button when hovered and <h2>'s when hovered */
        .accordion_arrow:hover,
        h2[data-collapsed="true"]:hover,
        h2[data-collapsed="false"]:hover {
            cursor: pointer;
        }
    }

    /* always hide accordion arrow button on print */
    @media only print {
        .accordion_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* tooltips plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* tooltip container */
        #tooltip {
            position: absolute;
            width: 50%;
            min-width: 240px;
            max-width: 75%;
            z-index: 1;
        }

        /* tooltip content */
        #tooltip_content {
            margin-bottom: 5px;
            padding: 20px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
            word-break: break-word;
        }

        /* tooltip copy of paragraphs and figures */
        #tooltip_content > p,
        #tooltip_content > figure {
            margin: 0;
            max-height: 320px;
            overflow-y: auto;
        }

        /* tooltip copy of <img> */
        #tooltip_content > figure > img {
            max-height: 260px;
        }

        /* navigation bar */
        #tooltip_nav_bar {
            margin-top: 10px;
            text-align: center;
        }

        /* navigation bar previous/next buton */
        #tooltip_nav_bar > .icon_button {
            position: relative;
            top: 3px;
        }

        /* navigation bar previous button */
        #tooltip_nav_bar > .icon_button:first-of-type {
            margin-right: 5px;
        }

        /* navigation bar next button */
        #tooltip_nav_bar > .icon_button:last-of-type {
            margin-left: 5px;
        }
    }

    /* always hide tooltip on print */
    @media only print {
        #tooltip {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* jump to first plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* jump button */
        .jump_arrow {
            position: relative;
            top: 0.125em;
            margin-right: 5px;
        }
    }

    /* always hide jump button on print */
    @media only print {
        .jump_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* link highlight plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anything with data-highlighted attribute true */
        [data-highlighted="true"] {
            background: #ffeb3b;
        }

        /* anything with data-selected attribute true */
        [data-selected="true"] {
            background: #ff8a65 !important;
        }

        /* animation definition for glow */
        @keyframes highlight_glow {
            0% {
                background: none;
            }
            10% {
                background: #bbdefb;
            }
            100% {
                background: none;
            }
        }

        /* anything with data-glow attribute true */
        [data-glow="true"] {
            animation: highlight_glow 2s;
        }
    }

    /* -------------------------------------------------- */
    /* table of contents plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* toc panel */
        #toc_panel {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            left: 0;
            background: #ffffff;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* toc panel when closed */
        #toc_panel[data-open="false"] {
            min-width: 60px;
            width: 60px;
            height: 60px;
            border-right: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc panel when open */
        #toc_panel[data-open="true"] {
            min-width: 260px;
            max-width: 480px;
            /* keep panel edge consistent distance away from "page" edge */
            width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
            bottom: 0;
            border-right: solid 1px #bdbdbd;
        }

        /* toc panel header */
        #toc_header {
            box-sizing: border-box;
            display: flex;
            flex-direction: row;
            align-items: center;
            height: 60px;
            margin: 0;
            padding: 20px;
        }

        /* toc panel header when hovered */
        #toc_header:hover {
            cursor: pointer;
        }

        /* toc panel header when panel open */
        #toc_panel[data-open="true"] > #toc_header {
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc open/close header button */
        #toc_button {
            margin-right: 20px;
        }

        /* hide toc list and header text when closed */
        #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
        #toc_panel[data-open="false"] > #toc_list {
            display: none;
        }

        /* toc list of entries */
        #toc_list {
            box-sizing: border-box;
            width: 100%;
            padding: 20px;
            position: absolute;
            top: calc(60px + 1px);
            bottom: 0;
            overflow: auto;
        }

        /* toc entry, link to section in document */
        .toc_link {
            display: block;
            padding: 5px;
            position: relative;
            font-weight: 600;
            text-decoration: none;
        }

        /* toc entry when hovered or when "viewed" */
        .toc_link:hover,
        .toc_link[data-viewing="true"] {
            background: #f5f5f5;
        }

        /* toc entry, level 1 indentation */
        .toc_link[data-level="1"] {
            margin-left: 0;
        }

        /* toc entry, level 2 indentation */
        .toc_link[data-level="2"] {
            margin-left: 20px;
        }

        /* toc entry, level 3 indentation */
        .toc_link[data-level="3"] {
            margin-left: 40px;
        }

        /* toc entry, level 4 indentation */
        .toc_link[data-level="4"] {
            margin-left: 60px;
        }

        /* toc entry bullets */
        #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
            position: absolute;
            left: -15px;
            top: -1px;
            font-size: 1.5em;
        }

        /* toc entry, level 2 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
            content: "\2022";
        }

        /* toc entry, level 3 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
            content: "\25AB";
        }

        /* toc entry, level 4 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
            content: "-";
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* push <body> ("page") element down to make room for toc icon */
        .toc_body_nudge {
            padding-top: 60px;
        }

        /* toc icon when panel closed and not hovered */
        #toc_panel[data-open="false"]:not(:hover) {
            background: rgba(255, 255, 255, 0.75);
        }
    }

    /* always hide toc panel on print */
    @media only print {
        #toc_panel {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* lightbox plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* regular <img> in document when hovered */
        .lightbox_document_img:hover {
            cursor: pointer;
        }

        .body_no_scroll {
            overflow: hidden !important;
        }

        /* screen overlay */
        #lightbox_overlay {
            display: flex;
            flex-direction: column;
            position: fixed;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            background: rgba(0, 0, 0, 0.75);
            z-index: 3;
        }

        /* middle area containing lightbox image */
        #lightbox_image_container {
            flex-grow: 1;
            display: flex;
            justify-content: center;
            align-items: center;
            overflow: hidden;
            position: relative;
            padding: 20px;
        }

        /* bottom area containing caption */
        #lightbox_bottom_container {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100px;
            min-height: 100px;
            max-height: 100px;
            background: rgba(0, 0, 0, 0.5);
        }

        /* image number info text box */
        #lightbox_number_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            left: 2px;
            top: 0;
            z-index: 4;
        }

        /* zoom info text box */
        #lightbox_zoom_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            right: 2px;
            top: 0;
            z-index: 4;
        }

        /* copy of image caption */
        #lightbox_caption {
            box-sizing: border-box;
            display: inline-block;
            width: 100%;
            max-height: 100%;
            padding: 10px 0;
            text-align: center;
            overflow-y: auto;
            color: #ffffff;
        }

        /* navigation previous/next button */
        .lightbox_button {
            width: 100px;
            height: 100%;
            min-width: 100px;
            min-height: 100%;
            color: #ffffff;
        }

        /* navigation previous/next button when hovered */
        .lightbox_button:hover {
            background: none !important;
        }

        /* navigation button icon */
        .lightbox_button > svg {
            height: 25px;
        }

        /* figure auto-number */
        #lightbox_caption > span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* lightbox image when hovered */
        #lightbox_img:hover {
            cursor: grab;
        }

        /* lightbox image when grabbed */
        #lightbox_img:active {
            cursor: grabbing;
        }
    }

    /* when on screen < 480px wide */
    @media only screen and (max-width: 480px) {
        /* make navigation buttons skinnier on small screens to make more room for caption text */
        .lightbox_button {
            width: 50px;
            min-width: 50px;
        }
    }

    /* always hide lightbox on print */
    @media only print {
        #lightbox_overlay {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* hypothesis (annotations) plugin */
    /* -------------------------------------------------- */

    /* side panel */
    .annotator-frame {
        width: 280px !important;
        z-index: 0 !important;
    }

    /* match highlight color to rest of theme */
    .annotator-highlights-always-on .annotator-hl {
        background-color: #ffeb3b !important;
    }

    /* match focused color to rest of theme */
    .annotator-hl.annotator-hl-focused {
        background-color: #ff8a65 !important;
    }

    /* match bucket bar color to rest of theme */
    .annotator-bucket-bar {
        background: #f5f5f5 !important;
    }

    /* always hide toolbar and tooltip on print */
    @media only print {
        .annotator-frame {
            display: none !important;
        }

        hypothesis-adder {
            display: none !important;
        }
    }
</style>
<!-- table scroll plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows tables that are too wide to fit within
        // the page to have a scrollbar instead of being squashed to fit.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableScroll';

        // default plugin options
        const options = {
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // wrap each table in a container
            const tables = document.querySelectorAll('table');
            for (const table of tables)
                wrapElement(table).classList.add('table_wrapper');
            // table_wrapper CSS class in theme file provides scroll
            // functionality
        }

        // wrap element in div and return div
        function wrapElement(element) {
            const parent = element.parentNode;
            const wrapper = document.createElement('div');
            parent.replaceChild(wrapper, element);
            wrapper.appendChild(element);
            return wrapper;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- anchors plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds an anchor next to each of a certain type
        // of element that provides a human-readable url to that specific
        // item/position in the document (eg "manuscript.html#abstract"). It
        // also makes it such that scrolling out of view of a target removes
        // its identifier from the url.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'anchors';

        // default plugin options
        const options = {
            // which types of elements to add anchors next to, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3, figure, table',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // add anchor to each element of specified types
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements)
                addAnchor(element);

            // attach scroll listener to window
            window.addEventListener('scroll', onScroll);
        }

        // when window is scrolled
        function onScroll() {
            // if url has hash and user has scrolled out of view of hash
            // target, remove hash from url
            const target = getHashTarget();
            if (target) {
                if (
                    target.getBoundingClientRect().top > window.innerHeight ||
                    target.getBoundingClientRect().bottom < 0
                )
                    history.pushState(null, null, ' ');
            }
        }

        // add anchor to element
        function addAnchor(element) {
            let withId; // element with unique id
            let addTo; // element to add anchor button to

            // if figure or table, modify withId and addTo to get expected
            // elements
            if (element.tagName.toLowerCase() === 'figure') {
                withId = element.querySelector('img');
                addTo = element.querySelector('figcaption');
            } else if (element.tagName.toLowerCase() === 'table') {
                withId =
                    element.previousElementSibling ||
                    element.parentNode.previousElementSibling;
                addTo = element.querySelector('caption');
            }

            withId = withId || element;
            addTo = addTo || element;
            const id = withId.id || withId.name || null;

            // do not add anchor if element doesn't have assigned id.
            // id is generated by pandoc and is assumed to be unique and
            // human-readable
            if (!id)
                return;

            // create anchor button
            const anchor = document.createElement('a');
            anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
            anchor.title = 'Link to this part of the document';
            anchor.classList.add('icon_button', 'anchor');
            anchor.dataset.ignore = 'true';
            anchor.href = '#' + id;
            addTo.appendChild(anchor);
        }

        // get element that is target of link or url hash
        function getHashTarget() {
            const hash = window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector(
                '[id="' + id + '"], [name="' + id + '"]'
            );
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (hash.indexOf('#fig:') === 0)
                target = target.parentNode;
            if (hash.indexOf('#tbl:') === 0)
                target = target.nextElementSibling;

            return target;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- link icon -->

<template class="icon_link">
    <!-- modified from: https://fontawesome.com/icons/link -->
    <svg width="16" height="16" viewBox="0 0 512 512">
        <path
            fill="currentColor"
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        ></path>
    </svg>
</template>
<!-- accordion plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows sections of content under <h2> headings
        // to be collapsible.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'accordion';

        // default plugin options
        const options = {
            // whether to always start expanded ('false'), always start
            // collapsed ('true'), or start collapsed when screen small ('auto')
            startCollapsed: 'auto',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <h2> heading
            const headings = document.querySelectorAll('h2');
            for (const heading of headings) {
                addArrow(heading);

                // start expanded/collapsed based on option
                if (
                    options.startCollapsed === 'true' ||
                    (options.startCollapsed === 'auto' && isSmallScreen())
                )
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // attach hash change listener to window
            window.addEventListener('hashchange', onHashChange);
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                goToElement(target);
        }

        // add arrow to heading
        function addArrow(heading) {
            // add arrow button
            const arrow = document.createElement('button');
            arrow.innerHTML = document.querySelector(
                '.icon_angle_down'
            ).innerHTML;
            arrow.classList.add('icon_button', 'accordion_arrow');
            heading.insertBefore(arrow, heading.firstChild);

            // attach click listener to heading and button
            heading.addEventListener('click', onHeadingClick);
            arrow.addEventListener('click', onArrowClick);
        }

        // determine if on mobile-like device with small screen
        function isSmallScreen() {
            return Math.min(window.innerWidth, window.innerHeight) < 480;
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get element that is target of hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector(
                '[id="' + id + '"], [name="' + id + '"]'
            );
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (hash.indexOf('#fig:') === 0)
                target = target.parentNode;
            if (hash.indexOf('#tbl:') === 0)
                target = target.nextElementSibling;

            return target;
        }

        // when <h2> heading is clicked
        function onHeadingClick(event) {
            // only collapse if <h2> itself is target of click (eg, user did
            // not click on anchor within <h2>)
            if (event.target === this)
                toggleCollapse(this);
        }

        // when arrow button is clicked
        function onArrowClick() {
            toggleCollapse(this.parentNode);
        }

        // collapse section if expanded, expand if collapsed
        function toggleCollapse(heading) {
            if (heading.dataset.collapsed === 'false')
                collapseHeading(heading);
            else
                expandHeading(heading);
        }

        // elements to exclude from collapse, such as table of contents panel,
        // hypothesis panel, etc
        const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

        // collapse section
        function collapseHeading(heading) {
            heading.setAttribute('data-collapsed', 'true');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'true');
        }

        // expand section
        function expandHeading(heading) {
            heading.setAttribute('data-collapsed', 'false');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'false');
        }

        // get list of elements between this <h2> and next <h2> or <h1>
        // ("children" of the <h2> section)
        function getChildren(heading) {
            return nextUntil(heading, 'h2, h1', exclude);
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get list of elements after a start element up to element matching
        // query
        function nextUntil(element, query, exclude) {
            const elements = [];
            while (element = element.nextElementSibling, element) {
                if (element.matches(query))
                    break;
                if (!element.matches(exclude))
                    elements.push(element);
            }
            return elements;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
    <!-- modified from: https://fontawesome.com/icons/angle-down -->
    <svg width="16" height="16" viewBox="0 0 448 512">
        <path
            fill="currentColor"
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        ></path>
    </svg>
</template>
<!-- tooltips plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when the user hovers or
        // focuses a link to a citation or figure, a tooltip appears with a
        // preview of the reference content, along with arrows to navigate
        // between instances of the same reference in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tooltips';

        // default plugin options
        const options = {
            // whether user must click off to close tooltip instead of just
            // un-hovering
            clickClose: 'false',
            // delay (in ms) between opening and closing tooltip
            delay: '100',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach hover and focus listeners to link
                link.addEventListener('mouseover', onLinkHover);
                link.addEventListener('mouseleave', onLinkUnhover);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('touchend', onLinkTouch);
            }

            // attach mouse, key, and resize listeners to window
            window.addEventListener('mousedown', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('keyup', onKeyUp);
            window.addEventListener('resize', onResize);
        }

        // when link is hovered
        function onLinkHover() {
            // function to open tooltip
            const delayOpenTooltip = function() {
                openTooltip(this);
            }.bind(this);

            // run open function after delay
            this.openTooltipTimer = window.setTimeout(
                delayOpenTooltip,
                options.delay
            );
        }

        // when mouse leaves link
        function onLinkUnhover() {
            // cancel opening tooltip
            window.clearTimeout(this.openTooltipTimer);

            // don't close on unhover if option specifies
            if (options.clickClose === 'true')
                return;

            // function to close tooltip
            const delayCloseTooltip = function() {
                // if tooltip open and if mouse isn't over tooltip, close
                const tooltip = document.getElementById('tooltip');
                if (tooltip && !tooltip.matches(':hover'))
                    closeTooltip();
            };

            // run close function after delay
            this.closeTooltipTimer = window.setTimeout(
                delayCloseTooltip,
                options.delay
            );
        }

        // when link is focused (tabbed to)
        function onLinkFocus(event) {
            openTooltip(this);
        }

        // when link is touched on touch screen
        function onLinkTouch(event) {
            // attempt to force hover state on first tap always, and trigger
            // regular link click (and navigation) on second tap
            if (event.target === document.activeElement)
                event.target.click();
            else {
                document.activeElement.blur();
                event.target.focus();
            }
            if (event.cancelable)
                event.preventDefault();
            event.stopPropagation();
            return false;
        }

        // when mouse is clicked anywhere in window
        function onClick(event) {
            closeTooltip();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'tooltip_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'tooltip_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeTooltip();
                    break;
            }
        }

        // when window is resized or zoomed
        function onResize() {
            closeTooltip();
        }

        // get all links of types we wish to handle
        function getLinks() {
            const queries = [];
            // exclude buttons, anchor links, toc links, etc
            const exclude =
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            queries.push('a[href^="#ref-"]' + exclude); // citation links
            queries.push('a[href^="#fig:"]' + exclude); // figure links
            const query = queries.join(', ');
            return document.querySelectorAll(query);
        }

        // get links with same target, get index of link in set, get total
        // same links
        function getSameLinks(link) {
            const sameLinks = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    sameLinks.push(otherLink);
            }

            return {
                elements: sameLinks,
                index: sameLinks.indexOf(link),
                total: sameLinks.length
            };
        }

        // open tooltip
        function openTooltip(link) {
            // delete tooltip if it exists, start fresh
            closeTooltip();

            // make tooltip element
            const tooltip = makeTooltip(link);

            // if source couldn't be found and tooltip not made, exit
            if (!tooltip)
                return;

            // make navbar elements
            const navBar = makeNavBar(link);
            if (navBar)
                tooltip.firstElementChild.appendChild(navBar);

            // attach tooltip to page
            document.body.appendChild(tooltip);

            // position tooltip
            const position = function() {
                positionTooltip(link);
            };
            position();

            // if tooltip contains images, position again after they've loaded
            const imgs = tooltip.querySelectorAll('img');
            for (const img of imgs)
                img.addEventListener('load', position);
        }

        // close (delete) tooltip
        function closeTooltip() {
            const tooltip = document.getElementById('tooltip');
            if (tooltip)
                tooltip.remove();
        }

        // make tooltip
        function makeTooltip(link) {
            // get target element that link points to
            const source = getSource(link);

            // if source can't be found, exit
            if (!source)
                return;

            // create new tooltip
            const tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            const tooltipContent = document.createElement('div');
            tooltipContent.id = 'tooltip_content';
            tooltip.appendChild(tooltipContent);

            // make copy of source node and put in tooltip
            const sourceCopy = makeCopy(source);
            tooltipContent.appendChild(sourceCopy);

            // attach mouse event listeners
            tooltip.addEventListener('click', onTooltipClick);
            tooltip.addEventListener('mousedown', onTooltipClick);
            tooltip.addEventListener('touchstart', onTooltipClick);
            tooltip.addEventListener('mouseleave', onTooltipUnhover);

            // (for interaction with lightbox plugin)
            // transfer click on tooltip copied img to original img
            const sourceImg = source.querySelector('img');
            const sourceCopyImg = sourceCopy.querySelector('img');
            if (sourceImg && sourceCopyImg) {
                const clickImg = function() {
                    sourceImg.click();
                    closeTooltip();
                };
                sourceCopyImg.addEventListener('click', clickImg);
            }

            return tooltip;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // when tooltip is clicked
        function onTooltipClick(event) {
            // when user clicks on tooltip, stop click from transferring
            // outside of tooltip (eg, click off to close tooltip, or eg click
            // off to unhighlight same refs)
            event.stopPropagation();
        }

        // when tooltip is unhovered
        function onTooltipUnhover(event) {
            if (options.clickClose === 'true')
                return;

            // make sure new mouse/touch/focus no longer over tooltip or any
            // element within it
            const tooltip = document.getElementById('tooltip');
            if (!tooltip)
                return;
            if (this.contains(event.relatedTarget))
                return;

            closeTooltip();
        }

        // make nav bar to go betwen prev/next instances of same reference
        function makeNavBar(link) {
            // find other links to the same source
            const sameLinks = getSameLinks(link);

            // don't show nav bar when singular reference
            if (sameLinks.total <= 1)
                return;

            // find prev/next links with same target
            const prevLink = getPrevLink(link, sameLinks);
            const nextLink = getNextLink(link, sameLinks);

            // create nav bar
            const navBar = document.createElement('div');
            navBar.id = 'tooltip_nav_bar';
            const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

            // create nav bar prev/next buttons
            const prevButton = document.createElement('button');
            const nextButton = document.createElement('button');
            prevButton.id = 'tooltip_prev_button';
            nextButton.id = 'tooltip_next_button';
            prevButton.title =
                'Jump to the previous occurence of this item in the document [←]';
            nextButton.title =
                'Jump to the next occurence of this item in the document [→]';
            prevButton.classList.add('icon_button');
            nextButton.classList.add('icon_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;
            navBar.appendChild(prevButton);
            navBar.appendChild(document.createTextNode(text));
            navBar.appendChild(nextButton);

            // attach click listeners to buttons
            prevButton.addEventListener('click', function() {
                onPrevNextClick(link, prevLink);
            });
            nextButton.addEventListener('click', function() {
                onPrevNextClick(link, nextLink);
            });

            return navBar;
        }

        // get previous link with same target
        function getPrevLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if < 1
            let index;
            if (sameLinks.index - 1 >= 0)
                index = sameLinks.index - 1;
            else
                index = sameLinks.total - 1;
            return sameLinks.elements[index];
        }

        // get next link with same target
        function getNextLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if > total
            let index;
            if (sameLinks.index + 1 <= sameLinks.total - 1)
                index = sameLinks.index + 1;
            else
                index = 0;
            return sameLinks.elements[index];
        }

        // get element that is target of link or url hash
        function getSource(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector(
                '[id="' + id + '"], [name="' + id + '"]'
            );
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (hash.indexOf('#ref-') === 0)
                target = target.querySelector('p');
            else if (hash.indexOf('#fig:') === 0)
                target = target.parentNode;
            else if (hash.indexOf('#tbl:') === 0)
                return;

            return target;
        }

        // when prev/next arrow button is clicked
        function onPrevNextClick(link, prevNextLink) {
            if (link && prevNextLink)
                goToElement(prevNextLink, window.innerHeight * 0.5);
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // determine position to place tooltip based on link position in
        // viewport and tooltip size
        function positionTooltip(link, left, top) {
            const tooltipElement = document.getElementById('tooltip');
            if (!tooltipElement)
                return;

            // get convenient vars for position/dimensions of
            // link/tooltip/page/view
            link = getRectInPage(link);
            const tooltip = getRectInPage(tooltipElement);
            const view = getRectInPage();

            // horizontal positioning
            if (left)
                // use explicit value
                left = left;
            else if (link.left + tooltip.width < view.right)
                // fit tooltip to right of link
                left = link.left;
            else if (link.right - tooltip.width > view.left)
                // fit tooltip to left of link
                left = link.right - tooltip.width;
            // center tooltip in view
            else
                left = (view.right - view.left) / 2 - tooltip.width / 2;

            // vertical positioning
            if (top)
                // use explicit value
                top = top;
            else if (link.top - tooltip.height > view.top)
                // fit tooltip above link
                top = link.top - tooltip.height;
            else if (link.bottom + tooltip.height < view.bottom)
                // fit tooltip below link
                top = link.bottom;
            else {
                // center tooltip in view
                top = view.top + view.height / 2 - tooltip.height / 2;
                // nudge off of link to left/right if possible
                if (link.right + tooltip.width < view.right)
                    left = link.right;
                else if (link.left - tooltip.width > view.left)
                    left = link.left - tooltip.width;
            }

            tooltipElement.style.left = left + 'px';
            tooltipElement.style.top = top + 'px';
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get position of element relative to page
        function getRectInPage(element) {
            const rect = getRectInView(element);
            const body = getRectInView(document.body);

            const newRect = {};
            newRect.left = rect.left - body.left;
            newRect.top = rect.top - body.top;
            newRect.right = rect.right - body.left;
            newRect.bottom = rect.bottom - body.top;
            newRect.width = rect.width;
            newRect.height = rect.height;

            return newRect;
        }

        // (for interaction with accordion plugin)
        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // (for interaction with accordion plugin)
        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- jump to first plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds a button next to each reference entry,
        // figure, and table that jumps the page to the first occurrence of a
        // link to that item in the manuscript.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'jumpToFirst';

        // default plugin options
        const options = {
            // whether to add buttons next to reference entries
            references: 'true',
            // whether to add buttons next to figures
            figures: 'true',
            // whether to add buttons next to tables
            tables: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            if (options.references !== 'false')
                makeReferenceButtons();
            if (options.figures !== 'false')
                makeFigureButtons();
            if (options.tables !== 'false')
                makeTableButtons();
        }

        // when jump button clicked
        function onButtonClick() {
            const first = getFirstOccurrence(this.dataset.id);
            if (!first)
                return;

            // update url hash so navigating "back" in history will return
            // user to jump button
            window.location.hash = this.dataset.id;
            // scroll to link
            window.setTimeout(function() {
                goToElement(first, window.innerHeight * 0.5);
            }, 0);
        }

        // get first occurence of link to item in document
        function getFirstOccurrence(id) {
            let query = 'a';
            query += '[href="#' + id + '"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelector(query);
        }

        // add button next to each reference entry
        function makeReferenceButtons() {
            const references = document.querySelectorAll('div[id^="ref-"]');
            for (const reference of references) {
                // get reference id and element to add button to
                const id = reference.id;
                const container = reference.firstElementChild;
                const first = getFirstOccurrence(id);

                // if can't find link to reference, ignore
                if (!first)
                    continue;

                // make jump button
                let button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this reference in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.innerHTML = button.outerHTML + container.innerHTML;
                button = container.firstElementChild;
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeFigureButtons() {
            const figures = document.querySelectorAll('img[id^="fig:"]');
            for (const figure of figures) {
                // get figure id and element to add button to
                const id = figure.id;
                const container = figure.nextElementSibling;
                const first = getFirstOccurrence(id);

                // if can't find link to figure, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this figure in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeTableButtons() {
            const tables = document.querySelectorAll('a[name^="tbl:"]');
            for (const table of tables) {
                // get ref id and element to add button to
                const id = table.name;
                const container = table.nextElementSibling.querySelector(
                    'caption'
                );
                const first = getFirstOccurrence(id);

                // if can't find link to table, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this table in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
    <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
    <svg width="16" height="16" viewBox="0 0 320 512">
        <path
            fill="currentColor"
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
        ></path>
    </svg>
</template>
<!-- link highlight plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user hovers or
        // focuses a link, other links that have the same target will be
        // highlighted. It also makes it such that when clicking a link, the
        // target of the link (eg reference, figure, table) is briefly
        // highlighted.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'linkHighlight';

        // default plugin options
        const options = {
            // whether to also highlight links that go to external urls
            externalLinks: 'false',
            // whether user must click off to unhighlight instead of just
            // un-hovering
            clickUnhighlight: 'false',
            // whether to also highlight links that are unique
            highlightUnique: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach mouse and focus listeners to link
                link.addEventListener('mouseenter', onLinkFocus);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('mouseleave', onLinkUnhover);
            }

            // attach click and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('hashchange', onHashChange);

            // run hash change on window load in case user has navigated
            // directly to hash
            onHashChange();
        }

        // when link is focused (tabbed to) or hovered
        function onLinkFocus() {
            highlight(this);
        }

        // when link is unhovered
        function onLinkUnhover() {
            if (options.clickUnhighlight !== 'true')
                unhighlightAll();
        }

        // when the mouse is clicked anywhere in window
        function onClick(event) {
            unhighlightAll();
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                glowElement(target);
        }

        // get element that is target of link or url hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector(
                '[id="' + id + '"], [name="' + id + '"]'
            );
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (hash.indexOf('#fig:') === 0)
                target = target.parentNode;
            else if (hash.indexOf('#tbl:') === 0)
                target = target.nextElementSibling.querySelector('caption');

            return target;
        }

        // start glow sequence on an element
        function glowElement(element) {
            const startGlow = function() {
                onGlowEnd();
                element.dataset.glow = 'true';
                element.addEventListener('animationend', onGlowEnd);
            };
            const onGlowEnd = function() {
                element.removeAttribute('data-glow');
                element.removeEventListener('animationend', onGlowEnd);
            };
            startGlow();
        }

        // highlight link and all others with same target
        function highlight(link) {
            // force unhighlight all to start fresh
            unhighlightAll();

            // get links with same target
            if (!link)
                return;
            const sameLinks = getSameLinks(link);

            // if link unique and option is off, exit and don't highlight
            if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                return;

            // highlight all same links, and "select" (special highlight) this
            // one
            for (const sameLink of sameLinks) {
                if (sameLink === link)
                    sameLink.setAttribute('data-selected', 'true');
                else
                    sameLink.setAttribute('data-highlighted', 'true');
            }
        }

        // unhighlight all links
        function unhighlightAll() {
            const links = getLinks();
            for (const link of links) {
                link.setAttribute('data-selected', 'false');
                link.setAttribute('data-highlighted', 'false');
            }
        }

        // get links with same target
        function getSameLinks(link) {
            const results = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    results.push(otherLink);
            }
            return results;
        }

        // get all links of types we wish to handle
        function getLinks() {
            let query = 'a';
            if (options.externalLinks !== 'true')
                query += '[href^="#"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelectorAll(query);
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- table of contents plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin provides a "table of contents" (toc) panel on
        // the side of the document that allows the user to conveniently
        // navigate between sections of the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableOfContents';

        // default plugin options
        const options = {
            // which types of elements to add links for, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3',
            // whether default behavior is to be closed ('false'), open
            // ('true'), or only open when screen wide enough to fit panel
            // ('auto'). note: still always starts closed when page loads.
            open: 'auto',
            // if list item is more than this many characters, text will be
            // truncated
            charLimit: '50',
            // whether or not to show bullets next to each toc item
            bullets: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // make toc panel and populate with entries (links to document
            // sections)
            const panel = makePanel();
            if (!panel)
                return;
            makeEntries(panel);
            document.body.insertBefore(panel, document.body.firstChild);

            closePanel();

            // attach click, scroll, and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('scroll', onScroll);
            window.addEventListener('hashchange', onScroll);
            window.addEventListener('keyup', onKeyUp);
            onScroll();

            // add class to push document body down out of way of toc button
            document.body.classList.add('toc_body_nudge');
        }

        // determine if screen wide enough to fit toc panel
        function isSmallScreen() {
            // in default theme:
            // 816px = 8.5in = width of "page" (<body>) element
            // 260px = min width of toc panel (*2 for both sides of <body>)
            return window.innerWidth < 816 + 260 * 2;
        }

        // open/close panel based on option and screen size
        function openOrClosePanel() {
            if (
                options.open === 'true' ||
                (options.open === 'auto' && !isSmallScreen())
            )
                openPanel();
            else
                closePanel();
        }

        // when mouse is clicked anywhere in window
        function onClick() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                openOrClosePanel();
        }

        // when window is scrolled or hash changed
        function onScroll() {
            highlightViewed();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            // close on esc
            if (event.key === 'Escape')
                closePanel();
        }

        // find entry of currently viewed document section in toc and highlight
        function highlightViewed() {
            const firstId = getFirstInView(options.typesQuery);

            // get toc entries (links), unhighlight all, then highlight viewed
            const list = document.getElementById('toc_list');
            if (!firstId || !list)
                return;
            const links = list.querySelectorAll('a');
            for (const link of links)
                link.dataset.viewing = 'false';
            const link = list.querySelector('a[href="#' + firstId + '"]');
            if (!link)
                return;
            link.dataset.viewing = 'true';
        }

        // get first or previous toc listed element in top half of view
        function getFirstInView(query) {
            // get all elements matching query and with id
            const elements = document.querySelectorAll(query);
            const elementsWithIds = [];
            for (const element of elements) {
                if (element.id)
                    elementsWithIds.push(element);
            }


            // get first or previous element in top half of view
            for (let i = 0; i < elementsWithIds.length; i++) {
                const element = elementsWithIds[i];
                const prevElement = elementsWithIds[Math.max(0, i - 1)];
                if (element.getBoundingClientRect().top >= 0) {
                    if (
                        element.getBoundingClientRect().top <
                        window.innerHeight / 2
                    )
                        return element.id;
                    else
                        return prevElement.id;
                }
            }
        }

        // make panel
        function makePanel() {
            // create panel
            const panel = document.createElement('div');
            panel.id = 'toc_panel';
            if (options.bullets === 'true')
                panel.dataset.bullets = 'true';

            // create header
            const header = document.createElement('div');
            header.id = 'toc_header';

            // create toc button
            const button = document.createElement('button');
            button.id = 'toc_button';
            button.innerHTML = document.querySelector(
                '.icon_th_list'
            ).innerHTML;
            button.classList.add('icon_button');

            // create header text
            const text = document.createElement('h3');
            text.innerHTML = 'Table of Contents';

            // create container for toc list
            const list = document.createElement('div');
            list.id = 'toc_list';

            // attach click listeners
            panel.addEventListener('click', onPanelClick);
            header.addEventListener('click', onHeaderClick);
            button.addEventListener('click', onButtonClick);

            // attach elements
            header.appendChild(button);
            header.appendChild(text);
            panel.appendChild(header);
            panel.appendChild(list);

            return panel;
        }

        // create toc entries (links) to each element of the specified types
        function makeEntries(panel) {
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements) {
                // do not add link if element doesn't have assigned id
                if (!element.id)
                    continue;

                // create link/list item
                const link = document.createElement('a');
                link.classList.add('toc_link');
                switch (element.tagName.toLowerCase()) {
                    case 'h1':
                        link.dataset.level = '1';
                        break;
                    case 'h2':
                        link.dataset.level = '2';
                        break;
                    case 'h3':
                        link.dataset.level = '3';
                        break;
                    case 'h4':
                        link.dataset.level = '4';
                        break;
                }
                link.title = element.innerText;
                let text = element.innerText;
                if (text.length > options.charLimit)
                    text = text.slice(0, options.charLimit) + '...';
                link.innerHTML = text;
                link.href = '#' + element.id;
                link.addEventListener('click', onLinkClick);

                // attach link
                panel.querySelector('#toc_list').appendChild(link);
            }
        }

        // when panel is clicked
        function onPanelClick(event) {
            // stop click from propagating to window/document and closing panel
            event.stopPropagation();
        }

        // when header itself is clicked
        function onHeaderClick(event) {
            togglePanel();
        }

        // when button is clicked
        function onButtonClick(event) {
            togglePanel();
            // stop header underneath button from also being clicked
            event.stopPropagation();
        }

        // when link is clicked
        function onLinkClick() {
            openOrClosePanel();
        }

        // open panel if closed, close if opened
        function togglePanel() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                closePanel();
            else
                openPanel();
        }

        // open panel
        function openPanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'true';
        }

        // close panel
        function closePanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'false';
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- th list icon -->

<template class="icon_th_list">
    <!-- modified from: https://fontawesome.com/icons/th-list -->
    <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
        <path
            fill="currentColor"
            d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- lightbox plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user clicks on an
        // image, the image fills the screen and the user can pan/drag/zoom
        // the image and navigate between other images in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'lightbox';

        // default plugin options
        const options = {
            // list of possible zoom/scale factors
            zoomSteps:
                '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
            // whether to fit image to view ('fit'), display at 100% and shrink
            // if necessary ('shrink'), or always display at 100% ('100')
            defaultZoom: 'fit',
            // whether to zoom in/out toward center of view ('true') or mouse
            // ('false')
            centerZoom: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <img> element
            const imgs = document.querySelectorAll('figure > img');
            let count = 1;
            for (const img of imgs) {
                img.classList.add('lightbox_document_img');
                img.dataset.number = count;
                img.dataset.total = imgs.length;
                img.addEventListener('click', openLightbox);
                count++;
            }

            // attach mouse and key listeners to window
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('keyup', onKeyUp);
        }

        // when mouse is moved anywhere in window
        function onWindowMouseMove(event) {
            window.mouseX = event.clientX;
            window.mouseY = event.clientY;
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'lightbox_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'lightbox_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeLightbox();
                    break;
            }
        }

        // open lightbox
        function openLightbox() {
            const lightbox = makeLightbox(this);
            if (!lightbox)
                return;

            blurBody(lightbox);
            document.body.appendChild(lightbox);
        }

        // make lightbox
        function makeLightbox(img) {
            // delete lightbox if it exists, start fresh
            closeLightbox();

            // create screen overlay containing lightbox
            const overlay = document.createElement('div');
            overlay.id = 'lightbox_overlay';

            // create image info boxes
            const numberInfo = document.createElement('div');
            const zoomInfo = document.createElement('div');
            numberInfo.id = 'lightbox_number_info';
            zoomInfo.id = 'lightbox_zoom_info';

            // create container for image
            const imageContainer = document.createElement('div');
            imageContainer.id = 'lightbox_image_container';
            const lightboxImg = makeLightboxImg(
                img,
                imageContainer,
                numberInfo,
                zoomInfo
            );
            imageContainer.appendChild(lightboxImg);

            // create bottom container for caption and navigation buttons
            const bottomContainer = document.createElement('div');
            bottomContainer.id = 'lightbox_bottom_container';
            const caption = makeCaption(img);
            const prevButton = makePrevButton(img);
            const nextButton = makeNextButton(img);
            bottomContainer.appendChild(prevButton);
            bottomContainer.appendChild(caption);
            bottomContainer.appendChild(nextButton);

            // attach top middle and bottom to overlay
            overlay.appendChild(numberInfo);
            overlay.appendChild(zoomInfo);
            overlay.appendChild(imageContainer);
            overlay.appendChild(bottomContainer);

            return overlay;
        }

        // make <img> object that is intuitively draggable and zoomable
        function makeLightboxImg(
            sourceImg,
            container,
            numberInfoBox,
            zoomInfoBox
        ) {
            // create copy of source <img>
            const img = sourceImg.cloneNode(true);
            img.classList.remove('lightbox_document_img');
            img.removeAttribute('id');
            img.removeAttribute('width');
            img.removeAttribute('height');
            img.style.position = 'unset';
            img.style.margin = '0';
            img.style.padding = '0';
            img.style.width = '';
            img.style.height = '';
            img.style.minWidth = '';
            img.style.minHeight = '';
            img.style.maxWidth = '';
            img.style.maxHeight = '';
            img.id = 'lightbox_img';

            // build sorted list of unique zoomSteps, always including a 100%
            let zoomSteps = [];
            const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
            for (const optionZoom of optionsZooms) {
                const newZoom = parseFloat(optionZoom);
                if (newZoom && !zoomSteps.includes(newZoom))
                    zoomSteps.push(newZoom);
            }
            if (!zoomSteps.includes(1))
                zoomSteps.push(1);
            zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                return a - b;
            });

            // <img> object property variables
            let zoom = 1;
            let translateX = 0;
            let translateY = 0;
            let clickMouseX = undefined;
            let clickMouseY = undefined;
            let clickTranslateX = undefined;
            let clickTranslateY = undefined;

            updateNumberInfo();

            // update image numbers displayed in info box
            function updateNumberInfo() {
                numberInfoBox.innerHTML =
                    sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
            }

            // update zoom displayed in info box
            function updateZoomInfo() {
                let zoomInfo = zoom * 100;
                if (!Number.isInteger(zoomInfo))
                    zoomInfo = zoomInfo.toFixed(2);
                zoomInfoBox.innerHTML = zoomInfo + '%';
            }

            // move to closest zoom step above current zoom
            const zoomIn = function() {
                for (const zoomStep of zoomSteps) {
                    if (zoomStep > zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                updateTransform();
            };

            // move to closest zoom step above current zoom
            const zoomOut = function() {
                zoomSteps.reverse();
                for (const zoomStep of zoomSteps) {
                    if (zoomStep < zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                zoomSteps.reverse();

                updateTransform();
            };

            // update display of <img> based on scale/translate properties
            const updateTransform = function() {
                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                // get new width/height after scale
                const rect = img.getBoundingClientRect();
                // limit translate
                translateX = Math.max(translateX, -rect.width / 2);
                translateX = Math.min(translateX, rect.width / 2);
                translateY = Math.max(translateY, -rect.height / 2);
                translateY = Math.min(translateY, rect.height / 2);

                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                updateZoomInfo();
            };

            // fit <img> to container
            const fit = function() {
                // no x/y offset, 100% zoom by default
                translateX = 0;
                translateY = 0;
                zoom = 1;

                // widths of <img> and container
                const imgWidth = img.naturalWidth;
                const imgHeight = img.naturalHeight;
                const containerWidth = parseFloat(
                    window.getComputedStyle(container).width
                );
                const containerHeight = parseFloat(
                    window.getComputedStyle(container).height
                );

                // how much zooming is needed to fit <img> to container
                const xRatio = imgWidth / containerWidth;
                const yRatio = imgHeight / containerHeight;
                const maxRatio = Math.max(xRatio, yRatio);
                const newZoom = 1 / maxRatio;

                // fit <img> to container according to option
                if (options.defaultZoom === 'shrink') {
                    if (maxRatio > 1)
                        zoom = newZoom;
                } else if (options.defaultZoom === 'fit')
                    zoom = newZoom;

                updateTransform();
            };

            // when mouse wheel is rolled anywhere in container
            const onContainerWheel = function(event) {
                if (!event)
                    return;

                // let ctrl + mouse wheel to zoom behave as normal
                if (event.ctrlKey)
                    return;

                // prevent normal scroll behavior
                event.preventDefault();
                event.stopPropagation();

                // point around which to scale img
                const viewRect = container.getBoundingClientRect();
                const viewX = (viewRect.left + viewRect.right) / 2;
                const viewY = (viewRect.top + viewRect.bottom) / 2;
                const originX = options.centerZoom === 'true' ? viewX : mouseX;
                const originY = options.centerZoom === 'true' ? viewY : mouseY;

                // get point on image under origin
                const oldRect = img.getBoundingClientRect();
                const oldPercentX = (originX - oldRect.left) / oldRect.width;
                const oldPercentY = (originY - oldRect.top) / oldRect.height;

                // increment/decrement zoom
                if (event.deltaY < 0)
                    zoomIn();
                if (event.deltaY > 0)
                    zoomOut();

                // get offset between previous image point and origin
                const newRect = img.getBoundingClientRect();
                const offsetX =
                    originX - (newRect.left + newRect.width * oldPercentX);
                const offsetY =
                    originY - (newRect.top + newRect.height * oldPercentY);

                // translate image to keep image point under origin
                translateX += offsetX;
                translateY += offsetY;

                // perform translate
                updateTransform();
            };

            // when container is clicked
            function onContainerClick(event) {
                // if container itself is target of click, and not other
                // element above it
                if (event.target === this)
                    closeLightbox();
            }

            // when mouse button is pressed on image
            const onImageMouseDown = function(event) {
                // store original mouse position relative to image
                clickMouseX = window.mouseX;
                clickMouseY = window.mouseY;
                clickTranslateX = translateX;
                clickTranslateY = translateY;
                event.stopPropagation();
                event.preventDefault();
            };

            // when mouse button is released anywhere in window
            const onWindowMouseUp = function(event) {
                // reset original mouse position
                clickMouseX = undefined;
                clickMouseY = undefined;
                clickTranslateX = undefined;
                clickTranslateY = undefined;

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mouseup', onWindowMouseUp);
            };

            // when mouse is moved anywhere in window
            const onWindowMouseMove = function(event) {
                if (
                    clickMouseX === undefined ||
                    clickMouseY === undefined ||
                    clickTranslateX === undefined ||
                    clickTranslateY === undefined
                )
                    return;

                // offset image based on original and current mouse position
                translateX = clickTranslateX + window.mouseX - clickMouseX;
                translateY = clickTranslateY + window.mouseY - clickMouseY;
                updateTransform();
                event.preventDefault();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mousemove', onWindowMouseMove);
            };

            // when window is resized
            const onWindowResize = function(event) {
                fit();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('resize', onWindowResize);
            };

            // attach the necessary event listeners
            img.addEventListener('dblclick', fit);
            img.addEventListener('mousedown', onImageMouseDown);
            container.addEventListener('wheel', onContainerWheel);
            container.addEventListener('mousedown', onContainerClick);
            container.addEventListener('touchstart', onContainerClick);
            window.addEventListener('mouseup', onWindowMouseUp);
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('resize', onWindowResize);

            // run fit() after lightbox atttached to document and <img> Loaded
            // so needed container and img dimensions available
            img.addEventListener('load', fit);

            return img;
        }

        // make caption
        function makeCaption(img) {
            const caption = document.createElement('div');
            caption.id = 'lightbox_caption';
            const captionSource = img.nextElementSibling;
            if (captionSource.tagName.toLowerCase() === 'figcaption') {
                const captionCopy = makeCopy(captionSource);
                caption.innerHTML = captionCopy.innerHTML;
            }

            caption.addEventListener('touchstart', function(event) {
                event.stopPropagation();
            });

            return caption;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // make button to jump to previous image in document
        function makePrevButton(img) {
            const prevButton = document.createElement('button');
            prevButton.id = 'lightbox_prev_button';
            prevButton.title = 'Jump to the previous image in the document [←]';
            prevButton.classList.add('icon_button', 'lightbox_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;

            // attach click listeners to button
            prevButton.addEventListener('click', function() {
                getPrevImg(img).click();
            });

            return prevButton;
        }

        // make button to jump to next image in document
        function makeNextButton(img) {
            const nextButton = document.createElement('button');
            nextButton.id = 'lightbox_next_button';
            nextButton.title = 'Jump to the next image in the document [→]';
            nextButton.classList.add('icon_button', 'lightbox_button');
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;

            // attach click listeners to button
            nextButton.addEventListener('click', function() {
                getNextImg(img).click();
            });

            return nextButton;
        }

        // get previous image in document
        function getPrevImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if < 1
            if (index - 1 >= 0)
                index--;
            else
                index = imgs.length - 1;
            return imgs[index];
        }

        // get next image in document
        function getNextImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if > total
            if (index + 1 <= imgs.length - 1)
                index++;
            else
                index = 0;
            return imgs[index];
        }

        // close lightbox
        function closeLightbox() {
            focusBody();

            const lightbox = document.getElementById('lightbox_overlay');
            if (lightbox)
                lightbox.remove();
        }

        // make all elements behind lightbox non-focusable
        function blurBody(overlay) {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.tabIndex = -1;
            document.body.classList.add('body_no_scroll');
        }

        // make all elements focusable again
        function focusBody() {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.removeAttribute('tabIndex');
            document.body.classList.remove('body_no_scroll');
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- math plugin configuration -->

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } }
    });
</script>

<!-- math plugin -->

<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
</script>
<!-- annotations plugin configuration -->

<script>
    window.hypothesisConfig = function() {
        return {
            branding: {
                accentColor: '#2196f3',
                appBackgroundColor: '#f8f8f8',
                ctaBackgroundColor: '#f8f8f8',
                ctaTextColor: '#000000',
                selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                annotationFontFamily: 'Open Sans, Helvetica, sans serif'
            }
        };
    };
</script>

<!-- annotations plugin -->

<script src='https://hypothes.is/embed.js'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'Hypothesis' allows public annotation of the
    // manuscript.

    // https://web.hypothes.is/
</script>
<!-- analytics plugin -->

<!-- copy and paste code from Google Analytics or similar service here -->
</body>
</html>
